{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/usr/bin/env python3\n",
    "##@author narumeena\n",
    "##@description slect only thoes variants that lie in coding region of genome "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import dask.dataframe as dd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>novelORF_ID</th>\n",
       "      <th>transcript_ID</th>\n",
       "      <th>transcriptBiotype</th>\n",
       "      <th>ORFannotation</th>\n",
       "      <th>inFrame</th>\n",
       "      <th>codingRegion</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>001aH1</td>\n",
       "      <td>ENST00000457540</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td>pseudogene</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>001bH1</td>\n",
       "      <td>ENST00000457540</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td>pseudogene</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>002kH1</td>\n",
       "      <td>ENST00000414273</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td>pseudogene</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>002kH2</td>\n",
       "      <td>ENST00000414273</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td>pseudogene</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>002nH1</td>\n",
       "      <td>ENST00000414273</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td>pseudogene</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  novelORF_ID    transcript_ID       transcriptBiotype ORFannotation  inFrame  \\\n",
       "0      001aH1  ENST00000457540  unprocessed_pseudogene    pseudogene    False   \n",
       "1      001bH1  ENST00000457540  unprocessed_pseudogene    pseudogene    False   \n",
       "2      002kH1  ENST00000414273  unprocessed_pseudogene    pseudogene    False   \n",
       "3      002kH2  ENST00000414273  unprocessed_pseudogene    pseudogene    False   \n",
       "4      002nH1  ENST00000414273  unprocessed_pseudogene    pseudogene    False   \n",
       "\n",
       "   codingRegion  \n",
       "0         False  \n",
       "1         False  \n",
       "2         False  \n",
       "3         False  \n",
       "4         False  "
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#classification of nORFs\n",
    "classified = pd.read_csv(\"../data/input/nORF/nORFclassification.tsv\",sep=\"\\t\")\n",
    "classified.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "233021"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(classified)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>novelORF_ID</th>\n",
       "      <th>transcript_ID</th>\n",
       "      <th>transcriptBiotype</th>\n",
       "      <th>ORFannotation</th>\n",
       "      <th>inFrame</th>\n",
       "      <th>codingRegion</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>00aeH2</td>\n",
       "      <td>ENST00000425657</td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>00ayH1</td>\n",
       "      <td>ENST00000342066</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>cds-intronic</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>00ayH2</td>\n",
       "      <td>ENST00000342066</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>cds</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>00bfH1</td>\n",
       "      <td>ENST00000341065</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>cds-intronic</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>00bfH2</td>\n",
       "      <td>ENST00000341065</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>cds</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   novelORF_ID    transcript_ID transcriptBiotype    ORFannotation  inFrame  \\\n",
       "11      00aeH2  ENST00000425657   retained_intron  retained_intron    False   \n",
       "13      00ayH1  ENST00000342066    protein_coding     cds-intronic    False   \n",
       "14      00ayH2  ENST00000342066    protein_coding              cds    False   \n",
       "15      00bfH1  ENST00000341065    protein_coding     cds-intronic    False   \n",
       "16      00bfH2  ENST00000341065    protein_coding              cds    False   \n",
       "\n",
       "    codingRegion  \n",
       "11          True  \n",
       "13          True  \n",
       "14          True  \n",
       "15          True  \n",
       "16          True  "
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "classified = classified[classified['codingRegion'] == True]\n",
    "classified.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "218838"
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(classified)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>novelORF_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>939111</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>00bfH1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>939111</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>00bfH2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>00eqH3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>00eqH5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>1022225</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>00ezH1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Chr      End Ref Alt novelORF_ID\n",
       "0   1   939111   C   T      00bfH1\n",
       "1   1   939111   C   T      00bfH2\n",
       "2   1  1014143   C   T      00eqH3\n",
       "3   1  1014143   C   T      00eqH5\n",
       "4   1  1022225   G   A      00ezH1"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#mapped variants to nORFs\n",
    "varinats_hgmd = pd.read_csv(\"../analysis/variantsMappedOnnORFs/hgmd/hgmd_mapped_nORF.txt\",sep=\"\\t\",header=None, converters={i: str for i in range(250)})\n",
    "varinats_hgmd = varinats_hgmd[[0,1,3,4,11]]\n",
    "varinats_hgmd.columns = ['Chr', 'End','Ref','Alt','novelORF_ID']\n",
    "varinats_hgmd.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>novelORF_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1014042</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>00eqH3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1014122</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>00eqH3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1014122</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>00eqH5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>00eqH3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>00eqH5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Chr      End Ref Alt novelORF_ID\n",
       "0   1  1014042   G   A      00eqH3\n",
       "1   1  1014122   C   T      00eqH3\n",
       "2   1  1014122   C   T      00eqH5\n",
       "3   1  1014143   C   T      00eqH3\n",
       "4   1  1014143   C   T      00eqH5"
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "varinats_clinvar = pd.read_csv(\"../analysis/variantsMappedOnnORFs/clinvar/clinvar_mapped_nORF.txt\",sep=\"\\t\",header=None, converters={i: str for i in range(250)})\n",
    "varinats_clinvar = varinats_clinvar[[0,1,3,4,11]]\n",
    "varinats_clinvar.columns = ['Chr', 'End','Ref','Alt','novelORF_ID']\n",
    "varinats_clinvar.head() \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>novelORF_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>918509</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1ihrH1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>924227</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>1ihvH2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>975043</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>00diH1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>999749</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1ikuH1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>1001225</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>00epH1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Chr      End Ref Alt novelORF_ID\n",
       "0   1   918509   A   G      1ihrH1\n",
       "1   1   924227   T   G      1ihvH2\n",
       "2   1   975043   T   C      00diH1\n",
       "3   1   999749   A   G      1ikuH1\n",
       "4   1  1001225   C   T      00epH1"
      ]
     },
     "execution_count": 78,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "varinats_human_drive = pd.read_csv(\"../analysis/variantsMappedOnnORFs/humanDrive/humanDrive_mapped_nORF.txt\",sep=\"\\t\",header=None, converters={i: str for i in range(250)})\n",
    "varinats_human_drive = varinats_human_drive[[0,1,3,4,8]]\n",
    "varinats_human_drive.columns = ['Chr', 'End','Ref','Alt','novelORF_ID']\n",
    "varinats_human_drive.head() \n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>novelORF_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>939111</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>00bfH1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>939111</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>00bfH2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>00eqH3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>00eqH5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>1022225</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>00ezH1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Chr      End Ref Alt novelORF_ID\n",
       "0   1   939111   C   T      00bfH1\n",
       "1   1   939111   C   T      00bfH2\n",
       "2   1  1014143   C   T      00eqH3\n",
       "3   1  1014143   C   T      00eqH5\n",
       "4   1  1022225   G   A      00ezH1"
      ]
     },
     "execution_count": 79,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#hgmd variants from coding region \n",
    "\n",
    "coding_variants_hgmd = pd.merge(classified, varinats_hgmd, how='inner',on=['novelORF_ID'])\n",
    "coding_variants_hgmd = coding_variants_hgmd[['Chr', 'End','Ref','Alt','novelORF_ID']] \n",
    "coding_variants_hgmd.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>novelORF_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1014042</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>00eqH3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1014122</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>00eqH3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>00eqH3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>1014179</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>00eqH3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>1014217</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>00eqH3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Chr      End Ref Alt novelORF_ID\n",
       "0   1  1014042   G   A      00eqH3\n",
       "1   1  1014122   C   T      00eqH3\n",
       "2   1  1014143   C   T      00eqH3\n",
       "3   1  1014179   C   T      00eqH3\n",
       "4   1  1014217   C   T      00eqH3"
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#clinvar variants from coding regions \n",
    "\n",
    "coding_variants_clinvar = pd.merge(classified, varinats_clinvar, how='inner',on=['novelORF_ID'])\n",
    "coding_variants_clinvar = coding_variants_clinvar[['Chr', 'End','Ref','Alt','novelORF_ID']] \n",
    "coding_variants_clinvar.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>975043</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1001225</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1001313</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>1001422</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>1001780</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Chr      End Ref Alt\n",
       "0   1   975043   T   C\n",
       "1   1  1001225   C   T\n",
       "2   1  1001313   G   T\n",
       "3   1  1001422   G   A\n",
       "4   1  1001780   G   A"
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#human drive from coding region and drop duplicates \n",
    "coding_variants_human_drive = pd.merge(classified, varinats_human_drive, how='inner',on=['novelORF_ID'])\n",
    "coding_variants_human_drive = coding_variants_human_drive[['Chr', 'End','Ref','Alt','novelORF_ID']] \n",
    "coding_variants_human_drive = coding_variants_human_drive[['Chr', 'End','Ref','Alt']].drop_duplicates()\n",
    "coding_variants_human_drive.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>939111</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1014143</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>1022225</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>1041582</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>1041648</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Chr      End Ref Alt\n",
       "0   1   939111   C   T\n",
       "2   1  1014143   C   T\n",
       "4   1  1022225   G   A\n",
       "5   1  1041582   C   T\n",
       "6   1  1041648   G   T"
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#concate two dataframe and drop duplicates \n",
    "coding_variants_hgmd_clinvar = pd.concat([coding_variants_hgmd,coding_variants_clinvar],axis=0, join='outer',ignore_index=True)\n",
    "coding_variants_hgmd_clinvar = coding_variants_hgmd_clinvar[['Chr', 'End','Ref','Alt']].drop_duplicates() \n",
    "coding_variants_hgmd_clinvar.head()\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>Length</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ConsScore</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>...</th>\n",
       "      <th>BP4</th>\n",
       "      <th>BP5</th>\n",
       "      <th>BP6</th>\n",
       "      <th>BP7</th>\n",
       "      <th>cosmic70</th>\n",
       "      <th>MCAP</th>\n",
       "      <th>REVEL</th>\n",
       "      <th>regsnp_fpr</th>\n",
       "      <th>regsnp_disease</th>\n",
       "      <th>regsnp_splicing_site</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1041582</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>STOP_GAINED</td>\n",
       "      <td>8</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1041582</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1042136</td>\n",
       "      <td>T</td>\n",
       "      <td>TC</td>\n",
       "      <td>INS</td>\n",
       "      <td>1</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>FRAME_SHIFT</td>\n",
       "      <td>7</td>\n",
       "      <td>frameshift</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>1051775</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>7</td>\n",
       "      <td>missense</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.628188449661</td>\n",
       "      <td>0.571</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>1051775</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>UPSTREAM</td>\n",
       "      <td>1</td>\n",
       "      <td>upstream</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>.</td>\n",
       "      <td>0.628188449661</td>\n",
       "      <td>0.571</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 270 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  Chr      End Ref Alt Type Length           AnnoType     Consequence  \\\n",
       "0   1  1041582   C   T  SNV      0   CodingTranscript     STOP_GAINED   \n",
       "1   1  1041582   C   T  SNV      0  RegulatoryFeature      REGULATORY   \n",
       "2   1  1042136   T  TC  INS      1   CodingTranscript     FRAME_SHIFT   \n",
       "3   1  1051775   G   A  SNV      0   CodingTranscript  NON_SYNONYMOUS   \n",
       "4   1  1051775   G   A  SNV      0         Intergenic        UPSTREAM   \n",
       "\n",
       "  ConsScore   ConsDetail  ...  BP4  BP5  BP6  BP7 cosmic70            MCAP  \\\n",
       "0         8  stop_gained  ...    0    0    0    0        .               .   \n",
       "1         4   regulatory  ...    0    0    0    0        .               .   \n",
       "2         7   frameshift  ...  NaN  NaN  NaN  NaN      NaN             NaN   \n",
       "3         7     missense  ...    0    0    0    0        .  0.628188449661   \n",
       "4         1     upstream  ...    0    0    0    0        .  0.628188449661   \n",
       "\n",
       "   REVEL regsnp_fpr regsnp_disease regsnp_splicing_site  \n",
       "0      .          .              .                    .  \n",
       "1      .          .              .                    .  \n",
       "2    NaN        NaN            NaN                  NaN  \n",
       "3  0.571          .              .                    .  \n",
       "4  0.571          .              .                    .  \n",
       "\n",
       "[5 rows x 270 columns]"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#attributes\n",
    "pathogenic = pd.read_csv(\"../analysis/trainingDataSet/pathogenic_with_all_attributes_cat_uniqe.tsv\",sep=\"\\t\", \n",
    "                         converters={i: str for i in range(250)})\n",
    "pathogenic.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [],
   "source": [
    "pathogenic = pathogenic.loc[pathogenic['Ref'].isin(['A','T','G','C'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [],
   "source": [
    "pathogenic = pathogenic.loc[pathogenic['Alt'].isin(['A','T','G','C'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [],
   "source": [
    "tbfs = dd.read_csv(\"../data/input/tbfs/snp2tfbs_JASPAR_CORE_2014_vertwithout_chr.bed\",sep=\"\\t\",header=None,\n",
    "                   names=[\"Chr\",\"Start\",\"End\",\"Ref\",\"Alt\",\"rsid\",\"tfbs\",\"tfbsName\",\"tbfsScore\"], converters={i: str for i in range(250)})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>rsid</th>\n",
       "      <th>tfbs</th>\n",
       "      <th>tfbsName</th>\n",
       "      <th>tbfsScore</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>10615</td>\n",
       "      <td>10616</td>\n",
       "      <td>CCGCCGTTGCAAAGGCGCGCCG</td>\n",
       "      <td>C</td>\n",
       "      <td>rs376342519</td>\n",
       "      <td>1</td>\n",
       "      <td>NRF1</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>11011</td>\n",
       "      <td>11012</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>rs544419019</td>\n",
       "      <td>2</td>\n",
       "      <td>PLAG1,EGR1</td>\n",
       "      <td>609,-186</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>13109</td>\n",
       "      <td>13110</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>rs540538026</td>\n",
       "      <td>3</td>\n",
       "      <td>ZNF263,PRDM1,SPIB</td>\n",
       "      <td>376,-9,0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>13115</td>\n",
       "      <td>13116</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>rs62635286</td>\n",
       "      <td>2</td>\n",
       "      <td>ZNF263,PRDM1</td>\n",
       "      <td>376,-9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>13117</td>\n",
       "      <td>13118</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>rs200579949</td>\n",
       "      <td>2</td>\n",
       "      <td>ZNF263,PRDM1</td>\n",
       "      <td>376,-9</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Chr  Start    End                     Ref Alt         rsid tfbs  \\\n",
       "0   1  10615  10616  CCGCCGTTGCAAAGGCGCGCCG   C  rs376342519    1   \n",
       "1   1  11011  11012                       C   G  rs544419019    2   \n",
       "2   1  13109  13110                       G   A  rs540538026    3   \n",
       "3   1  13115  13116                       T   G   rs62635286    2   \n",
       "4   1  13117  13118                       A   G  rs200579949    2   \n",
       "\n",
       "            tfbsName tbfsScore  \n",
       "0               NRF1        10  \n",
       "1         PLAG1,EGR1  609,-186  \n",
       "2  ZNF263,PRDM1,SPIB  376,-9,0  \n",
       "3       ZNF263,PRDM1    376,-9  \n",
       "4       ZNF263,PRDM1    376,-9  "
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tbfs.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1041582</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1041582</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>1051775</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>1051775</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>1233073</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Chr      End Ref Alt\n",
       "0   1  1041582   C   T\n",
       "1   1  1041582   C   T\n",
       "3   1  1051775   G   A\n",
       "4   1  1051775   G   A\n",
       "5   1  1233073   A   C"
      ]
     },
     "execution_count": 88,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pathogenic[[\"Chr\",\"End\",\"Ref\",\"Alt\"]].head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [],
   "source": [
    "pathogenic = dd.merge(pathogenic,tbfs, how='left', on=['Chr', 'End', 'Ref','Alt']).compute()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>Length</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ConsScore</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>...</th>\n",
       "      <th>MCAP</th>\n",
       "      <th>REVEL</th>\n",
       "      <th>regsnp_fpr</th>\n",
       "      <th>regsnp_disease</th>\n",
       "      <th>regsnp_splicing_site</th>\n",
       "      <th>Start_y</th>\n",
       "      <th>rsid</th>\n",
       "      <th>tfbs</th>\n",
       "      <th>tfbsName</th>\n",
       "      <th>tbfsScore</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1806503</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>7</td>\n",
       "      <td>missense</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0360599486135</td>\n",
       "      <td>0.515</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1806515</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>7</td>\n",
       "      <td>missense</td>\n",
       "      <td>...</td>\n",
       "      <td>0.653446970564</td>\n",
       "      <td>0.906</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>6040603</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>7</td>\n",
       "      <td>missense</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0635892354651</td>\n",
       "      <td>0.261</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>7962814</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>CodingTranscript</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>7</td>\n",
       "      <td>missense</td>\n",
       "      <td>...</td>\n",
       "      <td>0.311690713096</td>\n",
       "      <td>0.914</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>7962814</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>0.311690713096</td>\n",
       "      <td>0.914</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 275 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  Chr      End Ref Alt Type Length           AnnoType     Consequence  \\\n",
       "0   1  1806503   A   G  SNV      0   CodingTranscript  NON_SYNONYMOUS   \n",
       "1   1  1806515   T   C  SNV      0   CodingTranscript  NON_SYNONYMOUS   \n",
       "2   1  6040603   G   A  SNV      0   CodingTranscript  NON_SYNONYMOUS   \n",
       "3   1  7962814   T   C  SNV      0   CodingTranscript  NON_SYNONYMOUS   \n",
       "4   1  7962814   T   C  SNV      0  RegulatoryFeature      REGULATORY   \n",
       "\n",
       "  ConsScore  ConsDetail  ...             MCAP  REVEL regsnp_fpr  \\\n",
       "0         7    missense  ...  0.0360599486135  0.515          .   \n",
       "1         7    missense  ...   0.653446970564  0.906          .   \n",
       "2         7    missense  ...  0.0635892354651  0.261          .   \n",
       "3         7    missense  ...   0.311690713096  0.914          .   \n",
       "4         4  regulatory  ...   0.311690713096  0.914          .   \n",
       "\n",
       "  regsnp_disease regsnp_splicing_site Start_y rsid tfbs tfbsName tbfsScore  \n",
       "0              .                    .     NaN  NaN  NaN      NaN       NaN  \n",
       "1              .                    .     NaN  NaN  NaN      NaN       NaN  \n",
       "2              .                    .     NaN  NaN  NaN      NaN       NaN  \n",
       "3              .                    .     NaN  NaN  NaN      NaN       NaN  \n",
       "4              .                    .     NaN  NaN  NaN      NaN       NaN  \n",
       "\n",
       "[5 rows x 275 columns]"
      ]
     },
     "execution_count": 90,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding_pathogenic = dd.merge(pathogenic, coding_variants_hgmd_clinvar,how='inner', on=['Chr', 'End', 'Ref','Alt']) \n",
    "coding_pathogenic.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Chr',\n",
       " 'End',\n",
       " 'Ref',\n",
       " 'Alt',\n",
       " 'Type',\n",
       " 'Length',\n",
       " 'AnnoType',\n",
       " 'Consequence',\n",
       " 'ConsScore',\n",
       " 'ConsDetail',\n",
       " 'GC',\n",
       " 'CpG',\n",
       " 'motifECount',\n",
       " 'motifEName',\n",
       " 'motifEHIPos',\n",
       " 'motifEScoreChng',\n",
       " 'oAA',\n",
       " 'nAA',\n",
       " 'GeneID',\n",
       " 'FeatureID',\n",
       " 'GeneName',\n",
       " 'CCDS',\n",
       " 'Intron',\n",
       " 'Exon',\n",
       " 'cDNApos',\n",
       " 'relcDNApos',\n",
       " 'CDSpos',\n",
       " 'relCDSpos',\n",
       " 'protPos',\n",
       " 'relProtPos',\n",
       " 'Domain',\n",
       " 'Dst2Splice',\n",
       " 'Dst2SplType',\n",
       " 'minDistTSS',\n",
       " 'minDistTSE',\n",
       " 'SIFTcat',\n",
       " 'SIFTval',\n",
       " 'PolyPhenCat',\n",
       " 'PolyPhenVal',\n",
       " 'priPhCons',\n",
       " 'mamPhCons',\n",
       " 'verPhCons',\n",
       " 'priPhyloP',\n",
       " 'mamPhyloP',\n",
       " 'verPhyloP',\n",
       " 'bStatistic',\n",
       " 'targetScan',\n",
       " 'mirSVR-Score',\n",
       " 'mirSVR-E',\n",
       " 'mirSVR-Aln',\n",
       " 'cHmm_E1',\n",
       " 'cHmm_E2',\n",
       " 'cHmm_E3',\n",
       " 'cHmm_E4',\n",
       " 'cHmm_E5',\n",
       " 'cHmm_E6',\n",
       " 'cHmm_E7',\n",
       " 'cHmm_E8',\n",
       " 'cHmm_E9',\n",
       " 'cHmm_E10',\n",
       " 'cHmm_E11',\n",
       " 'cHmm_E12',\n",
       " 'cHmm_E13',\n",
       " 'cHmm_E14',\n",
       " 'cHmm_E15',\n",
       " 'cHmm_E16',\n",
       " 'cHmm_E17',\n",
       " 'cHmm_E18',\n",
       " 'cHmm_E19',\n",
       " 'cHmm_E20',\n",
       " 'cHmm_E21',\n",
       " 'cHmm_E22',\n",
       " 'cHmm_E23',\n",
       " 'cHmm_E24',\n",
       " 'cHmm_E25',\n",
       " 'GerpRS',\n",
       " 'GerpRSpval',\n",
       " 'GerpN',\n",
       " 'GerpS',\n",
       " 'tOverlapMotifs',\n",
       " 'motifDist',\n",
       " 'EncodeH3K4me1-sum',\n",
       " 'EncodeH3K4me1-max',\n",
       " 'EncodeH3K4me2-sum',\n",
       " 'EncodeH3K4me2-max',\n",
       " 'EncodeH3K4me3-sum',\n",
       " 'EncodeH3K4me3-max',\n",
       " 'EncodeH3K9ac-sum',\n",
       " 'EncodeH3K9ac-max',\n",
       " 'EncodeH3K9me3-sum',\n",
       " 'EncodeH3K9me3-max',\n",
       " 'EncodeH3K27ac-sum',\n",
       " 'EncodeH3K27ac-max',\n",
       " 'EncodeH3K27me3-sum',\n",
       " 'EncodeH3K27me3-max',\n",
       " 'EncodeH3K36me3-sum',\n",
       " 'EncodeH3K36me3-max',\n",
       " 'EncodeH3K79me2-sum',\n",
       " 'EncodeH3K79me2-max',\n",
       " 'EncodeH4K20me1-sum',\n",
       " 'EncodeH4K20me1-max',\n",
       " 'EncodeH2AFZ-sum',\n",
       " 'EncodeH2AFZ-max',\n",
       " 'EncodeDNase-sum',\n",
       " 'EncodeDNase-max',\n",
       " 'EncodetotalRNA-sum',\n",
       " 'EncodetotalRNA-max',\n",
       " 'Grantham',\n",
       " 'Dist2Mutation',\n",
       " 'Freq100bp',\n",
       " 'Rare100bp',\n",
       " 'Sngl100bp',\n",
       " 'Freq1000bp',\n",
       " 'Rare1000bp',\n",
       " 'Sngl1000bp',\n",
       " 'Freq10000bp',\n",
       " 'Rare10000bp',\n",
       " 'Sngl10000bp',\n",
       " 'EnsembleRegulatoryFeature',\n",
       " 'dbscSNV-ada_score',\n",
       " 'dbscSNV-rf_score',\n",
       " 'RemapOverlapTF',\n",
       " 'RemapOverlapCL',\n",
       " 'RawScore',\n",
       " 'PHRED',\n",
       " 'Start_x',\n",
       " 'Func.refGene',\n",
       " 'Gene.refGene',\n",
       " 'GeneDetail.refGene',\n",
       " 'ExonicFunc.refGene',\n",
       " 'AAChange.refGene',\n",
       " 'Func.knownGene',\n",
       " 'Gene.knownGene',\n",
       " 'GeneDetail.knownGene',\n",
       " 'ExonicFunc.knownGene',\n",
       " 'AAChange.knownGene',\n",
       " 'Func.ensGene',\n",
       " 'Gene.ensGene',\n",
       " 'GeneDetail.ensGene',\n",
       " 'ExonicFunc.ensGene',\n",
       " 'AAChange.ensGene',\n",
       " 'SIFT_score',\n",
       " 'SIFT_pred',\n",
       " 'Polyphen2_HDIV_score',\n",
       " 'Polyphen2_HDIV_pred',\n",
       " 'Polyphen2_HVAR_score',\n",
       " 'Polyphen2_HVAR_pred',\n",
       " 'LRT_score',\n",
       " 'LRT_pred',\n",
       " 'MutationTaster_score',\n",
       " 'MutationTaster_pred',\n",
       " 'MutationAssessor_score',\n",
       " 'MutationAssessor_pred',\n",
       " 'FATHMM_score',\n",
       " 'FATHMM_pred',\n",
       " 'RadialSVM_score',\n",
       " 'RadialSVM_pred',\n",
       " 'LR_score',\n",
       " 'LR_pred',\n",
       " 'VEST3_score',\n",
       " 'CADD_raw',\n",
       " 'CADD_phred',\n",
       " 'GERP++_RS',\n",
       " 'phyloP46way_placental',\n",
       " 'phyloP100way_vertebrate',\n",
       " 'SiPhy_29way_logOdds',\n",
       " 'Interpro_domain',\n",
       " 'SIFT_score.1',\n",
       " 'SIFT_converted_rankscore',\n",
       " 'SIFT_pred.1',\n",
       " 'Polyphen2_HDIV_score.1',\n",
       " 'Polyphen2_HDIV_rankscore',\n",
       " 'Polyphen2_HDIV_pred.1',\n",
       " 'Polyphen2_HVAR_score.1',\n",
       " 'Polyphen2_HVAR_rankscore',\n",
       " 'Polyphen2_HVAR_pred.1',\n",
       " 'LRT_score.1',\n",
       " 'LRT_converted_rankscore',\n",
       " 'LRT_pred.1',\n",
       " 'MutationTaster_score.1',\n",
       " 'MutationTaster_converted_rankscore',\n",
       " 'MutationTaster_pred.1',\n",
       " 'MutationAssessor_score.1',\n",
       " 'MutationAssessor_score_rankscore',\n",
       " 'MutationAssessor_pred.1',\n",
       " 'FATHMM_score.1',\n",
       " 'FATHMM_converted_rankscore',\n",
       " 'FATHMM_pred.1',\n",
       " 'PROVEAN_score',\n",
       " 'PROVEAN_converted_rankscore',\n",
       " 'PROVEAN_pred',\n",
       " 'VEST3_score.1',\n",
       " 'VEST3_rankscore',\n",
       " 'MetaSVM_score',\n",
       " 'MetaSVM_rankscore',\n",
       " 'MetaSVM_pred',\n",
       " 'MetaLR_score',\n",
       " 'MetaLR_rankscore',\n",
       " 'MetaLR_pred',\n",
       " 'M-CAP_score',\n",
       " 'M-CAP_rankscore',\n",
       " 'M-CAP_pred',\n",
       " 'CADD_raw.1',\n",
       " 'CADD_raw_rankscore',\n",
       " 'CADD_phred.1',\n",
       " 'DANN_score',\n",
       " 'DANN_rankscore',\n",
       " 'fathmm-MKL_coding_score',\n",
       " 'fathmm-MKL_coding_rankscore',\n",
       " 'fathmm-MKL_coding_pred',\n",
       " 'Eigen_coding_or_noncoding',\n",
       " 'Eigen-raw',\n",
       " 'Eigen-PC-raw',\n",
       " 'GenoCanyon_score',\n",
       " 'GenoCanyon_score_rankscore',\n",
       " 'integrated_fitCons_score',\n",
       " 'integrated_fitCons_score_rankscore',\n",
       " 'integrated_confidence_value',\n",
       " 'GERP++_RS.1',\n",
       " 'GERP++_RS_rankscore',\n",
       " 'phyloP100way_vertebrate.1',\n",
       " 'phyloP100way_vertebrate_rankscore',\n",
       " 'phyloP20way_mammalian',\n",
       " 'phyloP20way_mammalian_rankscore',\n",
       " 'phastCons100way_vertebrate',\n",
       " 'phastCons100way_vertebrate_rankscore',\n",
       " 'phastCons20way_mammalian',\n",
       " 'phastCons20way_mammalian_rankscore',\n",
       " 'SiPhy_29way_logOdds.1',\n",
       " 'SiPhy_29way_logOdds_rankscore',\n",
       " 'Interpro_domain.1',\n",
       " 'GTEx_V6_gene',\n",
       " 'GTEx_V6_tissue',\n",
       " 'dbscSNV_ADA_SCORE',\n",
       " 'dbscSNV_RF_SCORE',\n",
       " 'InterVar_automated',\n",
       " 'PVS1',\n",
       " 'PS1',\n",
       " 'PS2',\n",
       " 'PS3',\n",
       " 'PS4',\n",
       " 'PM1',\n",
       " 'PM2',\n",
       " 'PM3',\n",
       " 'PM4',\n",
       " 'PM5',\n",
       " 'PM6',\n",
       " 'PP1',\n",
       " 'PP2',\n",
       " 'PP3',\n",
       " 'PP4',\n",
       " 'PP5',\n",
       " 'BA1',\n",
       " 'BS1',\n",
       " 'BS2',\n",
       " 'BS3',\n",
       " 'BS4',\n",
       " 'BP1',\n",
       " 'BP2',\n",
       " 'BP3',\n",
       " 'BP4',\n",
       " 'BP5',\n",
       " 'BP6',\n",
       " 'BP7',\n",
       " 'cosmic70',\n",
       " 'MCAP',\n",
       " 'REVEL',\n",
       " 'regsnp_fpr',\n",
       " 'regsnp_disease',\n",
       " 'regsnp_splicing_site',\n",
       " 'Start_y',\n",
       " 'rsid',\n",
       " 'tfbs',\n",
       " 'tfbsName',\n",
       " 'tbfsScore']"
      ]
     },
     "execution_count": 91,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list(coding_pathogenic.columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>GC</th>\n",
       "      <th>CpG</th>\n",
       "      <th>motifECount</th>\n",
       "      <th>motifEHIPos</th>\n",
       "      <th>oAA</th>\n",
       "      <th>nAA</th>\n",
       "      <th>cDNApos</th>\n",
       "      <th>...</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>AAChange.knownGene</th>\n",
       "      <th>AAChange.ensGene</th>\n",
       "      <th>FATHMM_score</th>\n",
       "      <th>M-CAP_score</th>\n",
       "      <th>EnsembleRegulatoryFeature</th>\n",
       "      <th>dbscSNV-ada_score</th>\n",
       "      <th>dbscSNV-rf_score</th>\n",
       "      <th>RemapOverlapTF</th>\n",
       "      <th>tfbs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>0.503</td>\n",
       "      <td>0.04</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>I</td>\n",
       "      <td>T</td>\n",
       "      <td>583</td>\n",
       "      <td>...</td>\n",
       "      <td>GNB1:NM_001282539:exon5:c.T239C:p.I80T,GNB1:NM...</td>\n",
       "      <td>GNB1:uc031tmu.1:exon5:c.T239C:p.I80T,GNB1:uc05...</td>\n",
       "      <td>GNB1:ENST00000610897.4:exon5:c.T239C:p.I80T,GN...</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.036</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>0.497</td>\n",
       "      <td>0.04</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>D</td>\n",
       "      <td>G</td>\n",
       "      <td>571</td>\n",
       "      <td>...</td>\n",
       "      <td>GNB1:NM_001282539:exon5:c.A227G:p.D76G,GNB1:NM...</td>\n",
       "      <td>GNB1:uc031tmu.1:exon5:c.A227G:p.D76G,GNB1:uc05...</td>\n",
       "      <td>GNB1:ENST00000610897.4:exon5:c.A227G:p.D76G,GN...</td>\n",
       "      <td>-2.49</td>\n",
       "      <td>0.653</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>0.636</td>\n",
       "      <td>0.12</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>R</td>\n",
       "      <td>Q</td>\n",
       "      <td>520</td>\n",
       "      <td>...</td>\n",
       "      <td>KCNAB2:NM_001199860:exon2:c.G35A:p.R12Q,KCNAB2...</td>\n",
       "      <td>KCNAB2:uc001alu.4:exon2:c.G35A:p.R12Q,KCNAB2:u...</td>\n",
       "      <td>KCNAB2:ENST00000164247.5:exon2:c.G35A:p.R12Q,K...</td>\n",
       "      <td>0.84</td>\n",
       "      <td>0.064</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>0.411</td>\n",
       "      <td>0.027</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>L</td>\n",
       "      <td>P</td>\n",
       "      <td>96</td>\n",
       "      <td>...</td>\n",
       "      <td>PARK7:NM_001123377:exon2:c.T29C:p.L10P,PARK7:N...</td>\n",
       "      <td>PARK7:uc001aou.5:exon2:c.T29C:p.L10P,PARK7:uc0...</td>\n",
       "      <td>PARK7:ENST00000338639.9:exon2:c.T29C:p.L10P,PA...</td>\n",
       "      <td>-2.37</td>\n",
       "      <td>0.312</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>5</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>0.411</td>\n",
       "      <td>0.027</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>...</td>\n",
       "      <td>PARK7:NM_001123377:exon2:c.T29C:p.L10P,PARK7:N...</td>\n",
       "      <td>PARK7:uc001aou.5:exon2:c.T29C:p.L10P,PARK7:uc0...</td>\n",
       "      <td>PARK7:ENST00000338639.9:exon2:c.T29C:p.L10P,PA...</td>\n",
       "      <td>-2.37</td>\n",
       "      <td>0.312</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>5</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 103 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  Ref Alt     Consequence     GC    CpG motifECount motifEHIPos oAA nAA  \\\n",
       "0   A   G  NON_SYNONYMOUS  0.503   0.04          NA          NA   I   T   \n",
       "1   T   C  NON_SYNONYMOUS  0.497   0.04          NA          NA   D   G   \n",
       "2   G   A  NON_SYNONYMOUS  0.636   0.12          NA          NA   R   Q   \n",
       "3   T   C  NON_SYNONYMOUS  0.411  0.027          NA          NA   L   P   \n",
       "4   T   C      REGULATORY  0.411  0.027          NA          NA  NA  NA   \n",
       "\n",
       "  cDNApos  ...                                   AAChange.refGene  \\\n",
       "0     583  ...  GNB1:NM_001282539:exon5:c.T239C:p.I80T,GNB1:NM...   \n",
       "1     571  ...  GNB1:NM_001282539:exon5:c.A227G:p.D76G,GNB1:NM...   \n",
       "2     520  ...  KCNAB2:NM_001199860:exon2:c.G35A:p.R12Q,KCNAB2...   \n",
       "3      96  ...  PARK7:NM_001123377:exon2:c.T29C:p.L10P,PARK7:N...   \n",
       "4      NA  ...  PARK7:NM_001123377:exon2:c.T29C:p.L10P,PARK7:N...   \n",
       "\n",
       "                                  AAChange.knownGene  \\\n",
       "0  GNB1:uc031tmu.1:exon5:c.T239C:p.I80T,GNB1:uc05...   \n",
       "1  GNB1:uc031tmu.1:exon5:c.A227G:p.D76G,GNB1:uc05...   \n",
       "2  KCNAB2:uc001alu.4:exon2:c.G35A:p.R12Q,KCNAB2:u...   \n",
       "3  PARK7:uc001aou.5:exon2:c.T29C:p.L10P,PARK7:uc0...   \n",
       "4  PARK7:uc001aou.5:exon2:c.T29C:p.L10P,PARK7:uc0...   \n",
       "\n",
       "                                    AAChange.ensGene FATHMM_score M-CAP_score  \\\n",
       "0  GNB1:ENST00000610897.4:exon5:c.T239C:p.I80T,GN...         0.22       0.036   \n",
       "1  GNB1:ENST00000610897.4:exon5:c.A227G:p.D76G,GN...        -2.49       0.653   \n",
       "2  KCNAB2:ENST00000164247.5:exon2:c.G35A:p.R12Q,K...         0.84       0.064   \n",
       "3  PARK7:ENST00000338639.9:exon2:c.T29C:p.L10P,PA...        -2.37       0.312   \n",
       "4  PARK7:ENST00000338639.9:exon2:c.T29C:p.L10P,PA...        -2.37       0.312   \n",
       "\n",
       "  EnsembleRegulatoryFeature dbscSNV-ada_score dbscSNV-rf_score RemapOverlapTF  \\\n",
       "0                        NA                NA               NA              4   \n",
       "1                        NA                NA               NA              4   \n",
       "2                        NA                NA               NA              4   \n",
       "3                  Promoter                NA               NA              5   \n",
       "4                  Promoter                NA               NA              5   \n",
       "\n",
       "  tfbs  \n",
       "0  NaN  \n",
       "1  NaN  \n",
       "2  NaN  \n",
       "3  NaN  \n",
       "4  NaN  \n",
       "\n",
       "[5 rows x 103 columns]"
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "case = coding_pathogenic[[\"Ref\",\"Alt\",\"Consequence\",\"GC\",\"CpG\",\"motifECount\",\"motifEHIPos\",\"oAA\",\"nAA\",\"cDNApos\",\n",
    "           \"relcDNApos\",\"CDSpos\",\"relCDSpos\",\"protPos\",\"relProtPos\",\"Domain\",\"Dst2Splice\",\"Dst2SplType\",\n",
    "           \"minDistTSS\",\"minDistTSE\",\"SIFTcat\",\"SIFTval\",\"PolyPhenCat\",\"PolyPhenVal\",\"priPhCons\",\"mamPhCons\",\n",
    "           \"verPhCons\",\"priPhyloP\",\"mamPhyloP\",\"verPhyloP\",\"bStatistic\",\"cHmm_E1\",\"cHmm_E2\",\"cHmm_E3\",\"cHmm_E4\",\n",
    "           \"cHmm_E5\",\"cHmm_E6\",\"cHmm_E7\",\"cHmm_E8\",\"cHmm_E9\",\"cHmm_E10\",\"cHmm_E11\",\"cHmm_E12\",\"cHmm_E13\",\n",
    "           \"cHmm_E14\",\"cHmm_E15\",\"cHmm_E16\",\"cHmm_E17\",\"cHmm_E18\",\"cHmm_E19\",\"cHmm_E20\",\"cHmm_E21\",\"cHmm_E22\",\n",
    "           \"cHmm_E23\",\"cHmm_E24\",\"cHmm_E25\",\"GerpRS\",\"GerpRSpval\",\"GerpN\",\"GerpS\",\"tOverlapMotifs\",\"motifDist\",\n",
    "           \"EncodeH3K4me1-max\",\"EncodeH3K4me2-max\",\"EncodeH3K4me3-max\",\"EncodeH3K9ac-max\",\"EncodeH3K9me3-max\",\n",
    "           \"EncodeH3K27ac-max\",\"EncodeH3K27me3-max\",\"EncodeH3K36me3-max\",\"EncodeH3K79me2-max\",\"EncodeH4K20me1-max\",\n",
    "           \"EncodeH2AFZ-max\",\"EncodeDNase-max\",\"EncodetotalRNA-max\",\"Grantham\",\"SIFT_score\",\"Polyphen2_HDIV_score\",\n",
    "           \"Polyphen2_HVAR_score\",\"LRT_score\",\"MutationTaster_score\",\"MutationAssessor_score\",\"LR_score\",\"GERP++_RS\",\n",
    "           \"phyloP100way_vertebrate\",\"SiPhy_29way_logOdds\",\"PROVEAN_score\",\"MetaSVM_score\",\"MetaLR_score\",\n",
    "           \"integrated_fitCons_score\",\"phyloP20way_mammalian\",\"phastCons100way_vertebrate\",\n",
    "           \"phastCons20way_mammalian\",\"AAChange.refGene\",\"AAChange.knownGene\", \"AAChange.ensGene\",\n",
    "            \"FATHMM_score\",\"M-CAP_score\",\"EnsembleRegulatoryFeature\",\"dbscSNV-ada_score\",\n",
    "            \"dbscSNV-rf_score\",\"RemapOverlapTF\",\"tfbs\"]]\n",
    "\n",
    "#additional ,\"AAChange.refGene\",\"AAChange.knownGene\", \"AAChange.ensGene\",\"FATHMM_score\",\"M-CAP_score\",\"EnsembleRegulatoryFeature\",\"dbscSNV-ada_score\", \"dbscSNV-rf_score\",\"RemapOverlapTF\"\n",
    "\n",
    "\n",
    "case.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>GC</th>\n",
       "      <th>CpG</th>\n",
       "      <th>motifECount</th>\n",
       "      <th>motifEHIPos</th>\n",
       "      <th>oAA</th>\n",
       "      <th>nAA</th>\n",
       "      <th>cDNApos</th>\n",
       "      <th>...</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>AAChange.knownGene</th>\n",
       "      <th>AAChange.ensGene</th>\n",
       "      <th>FATHMM_score</th>\n",
       "      <th>M-CAP_score</th>\n",
       "      <th>EnsembleRegulatoryFeature</th>\n",
       "      <th>dbscSNV-ada_score</th>\n",
       "      <th>dbscSNV-rf_score</th>\n",
       "      <th>RemapOverlapTF</th>\n",
       "      <th>tfbs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>0.503</td>\n",
       "      <td>0.04</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>I</td>\n",
       "      <td>T</td>\n",
       "      <td>583</td>\n",
       "      <td>...</td>\n",
       "      <td>GNB1:NM_001282539:exon5:c.T239C:p.I80T,GNB1:NM...</td>\n",
       "      <td>GNB1:uc031tmu.1:exon5:c.T239C:p.I80T,GNB1:uc05...</td>\n",
       "      <td>GNB1:ENST00000610897.4:exon5:c.T239C:p.I80T,GN...</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.036</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>0.497</td>\n",
       "      <td>0.04</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>D</td>\n",
       "      <td>G</td>\n",
       "      <td>571</td>\n",
       "      <td>...</td>\n",
       "      <td>GNB1:NM_001282539:exon5:c.A227G:p.D76G,GNB1:NM...</td>\n",
       "      <td>GNB1:uc031tmu.1:exon5:c.A227G:p.D76G,GNB1:uc05...</td>\n",
       "      <td>GNB1:ENST00000610897.4:exon5:c.A227G:p.D76G,GN...</td>\n",
       "      <td>-2.49</td>\n",
       "      <td>0.653</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>0.636</td>\n",
       "      <td>0.12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>R</td>\n",
       "      <td>Q</td>\n",
       "      <td>520</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.84</td>\n",
       "      <td>0.064</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>0.411</td>\n",
       "      <td>0.027</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>L</td>\n",
       "      <td>P</td>\n",
       "      <td>96</td>\n",
       "      <td>...</td>\n",
       "      <td>PARK7:NM_001123377:exon2:c.T29C:p.L10P,PARK7:N...</td>\n",
       "      <td>PARK7:uc001aou.5:exon2:c.T29C:p.L10P,PARK7:uc0...</td>\n",
       "      <td>PARK7:ENST00000338639.9:exon2:c.T29C:p.L10P,PA...</td>\n",
       "      <td>-2.37</td>\n",
       "      <td>0.312</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>0.411</td>\n",
       "      <td>0.027</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>PARK7:NM_001123377:exon2:c.T29C:p.L10P,PARK7:N...</td>\n",
       "      <td>PARK7:uc001aou.5:exon2:c.T29C:p.L10P,PARK7:uc0...</td>\n",
       "      <td>PARK7:ENST00000338639.9:exon2:c.T29C:p.L10P,PA...</td>\n",
       "      <td>-2.37</td>\n",
       "      <td>0.312</td>\n",
       "      <td>Promoter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 103 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  Ref Alt     Consequence     GC    CpG motifECount motifEHIPos  oAA  nAA  \\\n",
       "0   A   G  NON_SYNONYMOUS  0.503   0.04         NaN         NaN    I    T   \n",
       "1   T   C  NON_SYNONYMOUS  0.497   0.04         NaN         NaN    D    G   \n",
       "2   G   A  NON_SYNONYMOUS  0.636   0.12         NaN         NaN    R    Q   \n",
       "3   T   C  NON_SYNONYMOUS  0.411  0.027         NaN         NaN    L    P   \n",
       "4   T   C      REGULATORY  0.411  0.027         NaN         NaN  NaN  NaN   \n",
       "\n",
       "  cDNApos  ...                                   AAChange.refGene  \\\n",
       "0     583  ...  GNB1:NM_001282539:exon5:c.T239C:p.I80T,GNB1:NM...   \n",
       "1     571  ...  GNB1:NM_001282539:exon5:c.A227G:p.D76G,GNB1:NM...   \n",
       "2     520  ...                                                NaN   \n",
       "3      96  ...  PARK7:NM_001123377:exon2:c.T29C:p.L10P,PARK7:N...   \n",
       "4     NaN  ...  PARK7:NM_001123377:exon2:c.T29C:p.L10P,PARK7:N...   \n",
       "\n",
       "                                  AAChange.knownGene  \\\n",
       "0  GNB1:uc031tmu.1:exon5:c.T239C:p.I80T,GNB1:uc05...   \n",
       "1  GNB1:uc031tmu.1:exon5:c.A227G:p.D76G,GNB1:uc05...   \n",
       "2                                                NaN   \n",
       "3  PARK7:uc001aou.5:exon2:c.T29C:p.L10P,PARK7:uc0...   \n",
       "4  PARK7:uc001aou.5:exon2:c.T29C:p.L10P,PARK7:uc0...   \n",
       "\n",
       "                                    AAChange.ensGene FATHMM_score M-CAP_score  \\\n",
       "0  GNB1:ENST00000610897.4:exon5:c.T239C:p.I80T,GN...         0.22       0.036   \n",
       "1  GNB1:ENST00000610897.4:exon5:c.A227G:p.D76G,GN...        -2.49       0.653   \n",
       "2                                                NaN         0.84       0.064   \n",
       "3  PARK7:ENST00000338639.9:exon2:c.T29C:p.L10P,PA...        -2.37       0.312   \n",
       "4  PARK7:ENST00000338639.9:exon2:c.T29C:p.L10P,PA...        -2.37       0.312   \n",
       "\n",
       "  EnsembleRegulatoryFeature dbscSNV-ada_score dbscSNV-rf_score RemapOverlapTF  \\\n",
       "0                       NaN               NaN              NaN              4   \n",
       "1                       NaN               NaN              NaN              4   \n",
       "2                       NaN               NaN              NaN              4   \n",
       "3                  Promoter               NaN              NaN              5   \n",
       "4                  Promoter               NaN              NaN              5   \n",
       "\n",
       "  tfbs  \n",
       "0  NaN  \n",
       "1  NaN  \n",
       "2  NaN  \n",
       "3  NaN  \n",
       "4  NaN  \n",
       "\n",
       "[5 rows x 103 columns]"
      ]
     },
     "execution_count": 93,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "case = case.replace('null', np.nan, regex=True)\n",
    "case = case.replace('Null', np.nan, regex=True)\n",
    "case = case.replace('NA', np.nan, regex=True)\n",
    "case = case.replace('.', np.nan)\n",
    "case = case.replace('-', np.nan)\n",
    "case = case.replace('*', np.nan)\n",
    "\n",
    "case.to_csv(\"../analysis/trainingDataSet/slected_pathogenic_Coding_region_only.csv\", sep=\"\\t\",index=False)\n",
    "\n",
    "case.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(77947, 103)"
      ]
     },
     "execution_count": 101,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "case.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>Length</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ConsScore</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>...</th>\n",
       "      <th>BP4</th>\n",
       "      <th>BP5</th>\n",
       "      <th>BP6</th>\n",
       "      <th>BP7</th>\n",
       "      <th>cosmic70</th>\n",
       "      <th>MCAP</th>\n",
       "      <th>REVEL</th>\n",
       "      <th>regsnp_fpr</th>\n",
       "      <th>regsnp_disease</th>\n",
       "      <th>regsnp_splicing_site</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>78208116</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>16</td>\n",
       "      <td>78215334</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>16</td>\n",
       "      <td>78220722</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>16</td>\n",
       "      <td>78221967</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>16</td>\n",
       "      <td>78221967</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 270 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  Chr       End Ref Alt Type Length           AnnoType Consequence ConsScore  \\\n",
       "0  16  78208116   G   A  SNV      0         Transcript    INTRONIC         2   \n",
       "1  16  78215334   T   C  SNV      0         Transcript    INTRONIC         2   \n",
       "2  16  78220722   A   G  SNV      0         Transcript    INTRONIC         2   \n",
       "3  16  78221967   G   C  SNV      0  RegulatoryFeature  REGULATORY         4   \n",
       "4  16  78221967   G   C  SNV      0         Transcript    INTRONIC         2   \n",
       "\n",
       "   ConsDetail  ... BP4 BP5 BP6 BP7 cosmic70 MCAP REVEL regsnp_fpr  \\\n",
       "0      intron  ...   .   .   .   .        .    .     .          .   \n",
       "1      intron  ...   .   .   .   .        .    .     .          .   \n",
       "2      intron  ...   .   .   .   .        .    .     .          .   \n",
       "3  regulatory  ...   .   .   .   .        .    .     .          .   \n",
       "4      intron  ...   .   .   .   .        .    .     .          .   \n",
       "\n",
       "  regsnp_disease regsnp_splicing_site  \n",
       "0              .                    .  \n",
       "1              .                    .  \n",
       "2              .                    .  \n",
       "3              .                    .  \n",
       "4              .                    .  \n",
       "\n",
       "[5 rows x 270 columns]"
      ]
     },
     "execution_count": 94,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "human_drive = pd.read_csv(\"../analysis/trainingDataSet/human_Drive_cadd_annovar_annotation_uniqe.tsv\",sep=\"\\t\", converters={i: str for i in range(250)})\n",
    "human_drive.head()\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [],
   "source": [
    "human_drive = human_drive.loc[human_drive['Ref'].isin(['A','T','G','C'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [],
   "source": [
    "human_drive = human_drive.loc[human_drive['Alt'].isin(['A','T','G','C'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [],
   "source": [
    "human_drive = dd.merge(human_drive,tbfs, how='left', on=['Chr', 'End', 'Ref','Alt']).compute()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>Length</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ConsScore</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>...</th>\n",
       "      <th>MCAP</th>\n",
       "      <th>REVEL</th>\n",
       "      <th>regsnp_fpr</th>\n",
       "      <th>regsnp_disease</th>\n",
       "      <th>regsnp_splicing_site</th>\n",
       "      <th>Start_y</th>\n",
       "      <th>rsid</th>\n",
       "      <th>tfbs</th>\n",
       "      <th>tfbsName</th>\n",
       "      <th>tbfsScore</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>78208116</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>16</td>\n",
       "      <td>78223503</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>16</td>\n",
       "      <td>78226080</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>16</td>\n",
       "      <td>78228288</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>16</td>\n",
       "      <td>78237299</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>1</td>\n",
       "      <td>downstream</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 275 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  Chr       End Ref Alt Type Length    AnnoType Consequence ConsScore  \\\n",
       "0  16  78208116   G   A  SNV      0  Transcript    INTRONIC         2   \n",
       "1  16  78223503   A   C  SNV      0  Transcript    INTRONIC         2   \n",
       "2  16  78226080   T   C  SNV      0  Transcript    INTRONIC         2   \n",
       "3  16  78228288   A   G  SNV      0  Transcript    INTRONIC         2   \n",
       "4  16  78237299   T   C  SNV      0  Intergenic  DOWNSTREAM         1   \n",
       "\n",
       "   ConsDetail  ... MCAP REVEL regsnp_fpr regsnp_disease regsnp_splicing_site  \\\n",
       "0      intron  ...    .     .          .              .                    .   \n",
       "1      intron  ...    .     .          .              .                    .   \n",
       "2      intron  ...    .     .          .              .                    .   \n",
       "3      intron  ...    .     .          .              .                    .   \n",
       "4  downstream  ...    .     .          .              .                    .   \n",
       "\n",
       "  Start_y rsid tfbs tfbsName tbfsScore  \n",
       "0     NaN  NaN  NaN      NaN       NaN  \n",
       "1     NaN  NaN  NaN      NaN       NaN  \n",
       "2     NaN  NaN  NaN      NaN       NaN  \n",
       "3     NaN  NaN  NaN      NaN       NaN  \n",
       "4     NaN  NaN  NaN      NaN       NaN  \n",
       "\n",
       "[5 rows x 275 columns]"
      ]
     },
     "execution_count": 98,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "coding_human_drive = dd.merge(human_drive, coding_variants_human_drive,how='inner', on=['Chr', 'End', 'Ref','Alt']) \n",
    "coding_human_drive.head()\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>GC</th>\n",
       "      <th>CpG</th>\n",
       "      <th>motifECount</th>\n",
       "      <th>motifEHIPos</th>\n",
       "      <th>oAA</th>\n",
       "      <th>nAA</th>\n",
       "      <th>cDNApos</th>\n",
       "      <th>...</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>AAChange.knownGene</th>\n",
       "      <th>AAChange.ensGene</th>\n",
       "      <th>FATHMM_score</th>\n",
       "      <th>M-CAP_score</th>\n",
       "      <th>EnsembleRegulatoryFeature</th>\n",
       "      <th>dbscSNV-ada_score</th>\n",
       "      <th>dbscSNV-rf_score</th>\n",
       "      <th>RemapOverlapTF</th>\n",
       "      <th>tfbs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>0.503</td>\n",
       "      <td>0.04</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>0.53</td>\n",
       "      <td>0.013</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>7</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>0.371</td>\n",
       "      <td>0.04</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>9</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>0.391</td>\n",
       "      <td>0.013</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>0.311</td>\n",
       "      <td>0.013</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 103 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  Ref Alt Consequence     GC    CpG motifECount motifEHIPos oAA nAA cDNApos  \\\n",
       "0   G   A    INTRONIC  0.503   0.04          NA          NA  NA  NA      NA   \n",
       "1   A   C    INTRONIC   0.53  0.013          NA          NA  NA  NA      NA   \n",
       "2   T   C    INTRONIC  0.371   0.04          NA          NA  NA  NA      NA   \n",
       "3   A   G    INTRONIC  0.391  0.013          NA          NA  NA  NA      NA   \n",
       "4   T   C  DOWNSTREAM  0.311  0.013          NA          NA  NA  NA      NA   \n",
       "\n",
       "   ... AAChange.refGene AAChange.knownGene AAChange.ensGene FATHMM_score  \\\n",
       "0  ...                .                  .                .            .   \n",
       "1  ...                .                  .                .            .   \n",
       "2  ...                .                  .                .            .   \n",
       "3  ...                .                  .                .            .   \n",
       "4  ...                .                  .                .            .   \n",
       "\n",
       "  M-CAP_score EnsembleRegulatoryFeature dbscSNV-ada_score dbscSNV-rf_score  \\\n",
       "0           .                        NA                NA               NA   \n",
       "1           .                        NA                NA               NA   \n",
       "2           .                        NA                NA               NA   \n",
       "3           .                        NA                NA               NA   \n",
       "4           .                        NA                NA               NA   \n",
       "\n",
       "  RemapOverlapTF tfbs  \n",
       "0             NA  NaN  \n",
       "1              7  NaN  \n",
       "2              9  NaN  \n",
       "3              2  NaN  \n",
       "4              2  NaN  \n",
       "\n",
       "[5 rows x 103 columns]"
      ]
     },
     "execution_count": 99,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "control = coding_human_drive[[\"Ref\",\"Alt\",\"Consequence\",\"GC\",\"CpG\",\"motifECount\",\"motifEHIPos\",\"oAA\",\"nAA\",\"cDNApos\",\n",
    "           \"relcDNApos\",\"CDSpos\",\"relCDSpos\",\"protPos\",\"relProtPos\",\"Domain\",\"Dst2Splice\",\"Dst2SplType\",\n",
    "           \"minDistTSS\",\"minDistTSE\",\"SIFTcat\",\"SIFTval\",\"PolyPhenCat\",\"PolyPhenVal\",\"priPhCons\",\"mamPhCons\",\n",
    "           \"verPhCons\",\"priPhyloP\",\"mamPhyloP\",\"verPhyloP\",\"bStatistic\",\"cHmm_E1\",\"cHmm_E2\",\"cHmm_E3\",\"cHmm_E4\",\n",
    "           \"cHmm_E5\",\"cHmm_E6\",\"cHmm_E7\",\"cHmm_E8\",\"cHmm_E9\",\"cHmm_E10\",\"cHmm_E11\",\"cHmm_E12\",\"cHmm_E13\",\n",
    "           \"cHmm_E14\",\"cHmm_E15\",\"cHmm_E16\",\"cHmm_E17\",\"cHmm_E18\",\"cHmm_E19\",\"cHmm_E20\",\"cHmm_E21\",\"cHmm_E22\",\n",
    "           \"cHmm_E23\",\"cHmm_E24\",\"cHmm_E25\",\"GerpRS\",\"GerpRSpval\",\"GerpN\",\"GerpS\",\"tOverlapMotifs\",\"motifDist\",\n",
    "           \"EncodeH3K4me1-max\",\"EncodeH3K4me2-max\",\"EncodeH3K4me3-max\",\"EncodeH3K9ac-max\",\"EncodeH3K9me3-max\",\n",
    "           \"EncodeH3K27ac-max\",\"EncodeH3K27me3-max\",\"EncodeH3K36me3-max\",\"EncodeH3K79me2-max\",\"EncodeH4K20me1-max\",\n",
    "           \"EncodeH2AFZ-max\",\"EncodeDNase-max\",\"EncodetotalRNA-max\",\"Grantham\",\"SIFT_score\",\"Polyphen2_HDIV_score\",\n",
    "           \"Polyphen2_HVAR_score\",\"LRT_score\",\"MutationTaster_score\",\"MutationAssessor_score\",\"LR_score\",\"GERP++_RS\",\n",
    "           \"phyloP100way_vertebrate\",\"SiPhy_29way_logOdds\",\"PROVEAN_score\",\"MetaSVM_score\",\"MetaLR_score\",\n",
    "           \"integrated_fitCons_score\",\"phyloP20way_mammalian\",\"phastCons100way_vertebrate\",\n",
    "           \"phastCons20way_mammalian\",\"AAChange.refGene\",\"AAChange.knownGene\", \"AAChange.ensGene\",\n",
    "            \"FATHMM_score\",\"M-CAP_score\",\"EnsembleRegulatoryFeature\",\"dbscSNV-ada_score\",\n",
    "            \"dbscSNV-rf_score\",\"RemapOverlapTF\",\"tfbs\"\n",
    "]]\n",
    "\n",
    "\n",
    "control.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>GC</th>\n",
       "      <th>CpG</th>\n",
       "      <th>motifECount</th>\n",
       "      <th>motifEHIPos</th>\n",
       "      <th>oAA</th>\n",
       "      <th>nAA</th>\n",
       "      <th>cDNApos</th>\n",
       "      <th>...</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>AAChange.knownGene</th>\n",
       "      <th>AAChange.ensGene</th>\n",
       "      <th>FATHMM_score</th>\n",
       "      <th>M-CAP_score</th>\n",
       "      <th>EnsembleRegulatoryFeature</th>\n",
       "      <th>dbscSNV-ada_score</th>\n",
       "      <th>dbscSNV-rf_score</th>\n",
       "      <th>RemapOverlapTF</th>\n",
       "      <th>tfbs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>0.503</td>\n",
       "      <td>0.04</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>0.53</td>\n",
       "      <td>0.013</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>0.371</td>\n",
       "      <td>0.04</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>0.391</td>\n",
       "      <td>0.013</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>0.311</td>\n",
       "      <td>0.013</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 103 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  Ref Alt Consequence     GC    CpG motifECount motifEHIPos  oAA  nAA cDNApos  \\\n",
       "0   G   A    INTRONIC  0.503   0.04         NaN         NaN  NaN  NaN     NaN   \n",
       "1   A   C    INTRONIC   0.53  0.013         NaN         NaN  NaN  NaN     NaN   \n",
       "2   T   C    INTRONIC  0.371   0.04         NaN         NaN  NaN  NaN     NaN   \n",
       "3   A   G    INTRONIC  0.391  0.013         NaN         NaN  NaN  NaN     NaN   \n",
       "4   T   C  DOWNSTREAM  0.311  0.013         NaN         NaN  NaN  NaN     NaN   \n",
       "\n",
       "   ... AAChange.refGene AAChange.knownGene AAChange.ensGene FATHMM_score  \\\n",
       "0  ...              NaN                NaN              NaN          NaN   \n",
       "1  ...              NaN                NaN              NaN          NaN   \n",
       "2  ...              NaN                NaN              NaN          NaN   \n",
       "3  ...              NaN                NaN              NaN          NaN   \n",
       "4  ...              NaN                NaN              NaN          NaN   \n",
       "\n",
       "  M-CAP_score EnsembleRegulatoryFeature dbscSNV-ada_score dbscSNV-rf_score  \\\n",
       "0         NaN                       NaN               NaN              NaN   \n",
       "1         NaN                       NaN               NaN              NaN   \n",
       "2         NaN                       NaN               NaN              NaN   \n",
       "3         NaN                       NaN               NaN              NaN   \n",
       "4         NaN                       NaN               NaN              NaN   \n",
       "\n",
       "  RemapOverlapTF tfbs  \n",
       "0            NaN  NaN  \n",
       "1              7  NaN  \n",
       "2              9  NaN  \n",
       "3              2  NaN  \n",
       "4              2  NaN  \n",
       "\n",
       "[5 rows x 103 columns]"
      ]
     },
     "execution_count": 100,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "control = control.replace('null', np.nan, regex=True)\n",
    "control = control.replace('Null', np.nan, regex=True)\n",
    "control = control.replace('NA', np.nan, regex=True)\n",
    "control = control.replace('.', np.nan)\n",
    "control = control.replace('-', np.nan)\n",
    "control = control.replace('*', np.nan)\n",
    "control.to_csv(\"../analysis/trainingDataSet/slected_human_Drive_Coding_region_only.csv\", sep=\"\\t\",index=False)\n",
    "control.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1515774, 103)"
      ]
     },
     "execution_count": 102,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "control.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
