{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {
    "code_folding": []
   },
   "outputs": [],
   "source": [
    "#!/usr/bin/env python3\n",
    "##@author narumeena\n",
    "##@description creating a prarallel cordinates plot in plotely \n",
    "\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import dask.dataframe as dd\n",
    "from matplotlib import pyplot as plt\n",
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {
    "code_folding": [],
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "\n",
    "pathogenic = dd.read_csv(\"../analysis/trainingDataSet/pathogenic_with_all_attributes_cat_uniqe.tsv\",sep=\"\\t\", \n",
    "                         converters={i: str for i in range(250)})\n",
    "#\"Chr\",\"End\",\"Ref\",\"Alt\",\"Type\",\"Length\",\"AnnoType\",\"Consequence\",\"ConsScore\",\"ConsDetail\",\"GC\",\"CpG\",\"motifECount\",\"motifEName\",\"motifEHIPos\",\"motifEScoreChng\",\"oAA\",\"nAA\",\"GeneID\",\"FeatureID\",\"GeneName\",\"CCDS\",\"Intron\",\"Exon\",\"cDNApos\",\"relcDNApos\",\"CDSpos\",\"relCDSpos\",\"protPos\",\"relProtPos\",\"Domain\",\"Dst2Splice\",\"Dst2SplType\",\"minDistTSS\",\"minDistTSE\",\"SIFTcat\",\"SIFTval\",\"PolyPhenCat\",\"PolyPhenVal\",\"priPhCons\",\"mamPhCons\",\"verPhCons\",\"priPhyloP\",\"mamPhyloP\",\"verPhyloP\",\"bStatistic\",\"targetScan\",\"mirSVR-Score\",\"mirSVR-E\",\"mirSVR-Aln\",\"cHmm_E1\",\"cHmm_E2\",\"cHmm_E3\",\"cHmm_E4\",\"cHmm_E5\",\"cHmm_E6\",\"cHmm_E7\",\"cHmm_E8\",\"cHmm_E9\",\"cHmm_E10\",\"cHmm_E11\",\"cHmm_E12\",\"cHmm_E13\",\"cHmm_E14\",\"cHmm_E15\",\"cHmm_E16\",\"cHmm_E17\",\"cHmm_E18\",\"cHmm_E19\",\"cHmm_E20\",\"cHmm_E21\",\"cHmm_E22\",\"cHmm_E23\",\"cHmm_E24\",\"cHmm_E25\",\"GerpRS\",\"GerpRSpval\",\"GerpN\",\"GerpS\",\"tOverlapMotifs\",\"motifDist\",\"EncodeH3K4me1-sum\",\"EncodeH3K4me1-max\",\"EncodeH3K4me2-sum\",\"EncodeH3K4me2-max\",\"EncodeH3K4me3-sum\",\"EncodeH3K4me3-max\",\"EncodeH3K9ac-sum\",\"EncodeH3K9ac-max\",\"EncodeH3K9me3-sum\",\"EncodeH3K9me3-max\",\"EncodeH3K27ac-sum\",\"EncodeH3K27ac-max\",\"EncodeH3K27me3-sum\",\"EncodeH3K27me3-max\",\"EncodeH3K36me3-sum\",\"EncodeH3K36me3-max\",\"EncodeH3K79me2-sum\",\"EncodeH3K79me2-max\",\"EncodeH4K20me1-sum\",\"EncodeH4K20me1-max\",\"EncodeH2AFZ-sum\",\"EncodeH2AFZ-max\",\"EncodeDNase-sum\",\"EncodeDNase-max\",\"EncodetotalRNA-sum\",\"EncodetotalRNA-max\",\"Grantham\",\"Dist2Mutation\",\"Freq100bp\",\"Rare100bp\",\"Sngl100bp\",\"Freq1000bp\",\"Rare1000bp\",\"Sngl1000bp\",\"Freq10000bp\",\"Rare10000bp\",\"Sngl10000bp\",\"EnsembleRegulatoryFeature\",\"dbscSNV-ada_score\",\"dbscSNV-rf_score\",\"RemapOverlapTF\",\"RemapOverlapCRawScore\",\"PHRED\",\"Start\",\"Func.refGene\",\"Gene.refGene\",\"GeneDetail.refGene\",\"ExonicFunc.refGene\",\"AAChange.refGene\",\"Func.knownGene\",\"Gene.knownGene\",\"GeneDetail.knownGene\",\"ExonicFunc.knownGene\",\"AAChange.knownGene\",\"Func.ensGene\",\"Gene.ensGene\",\"GeneDetail.ensGene\",\"ExonicFunc.ensGene\",\"AAChange.ensGene\",\"SIFT_score\",\"SIFT_pred\",\"Polyphen2_HDIV_score\",\"Polyphen2_HDIV_pred\",\"Polyphen2_HVAR_score\",\"Polyphen2_HVAR_pred\",\"LRT_score\",\"LRT_pred\",\"MutationTaster_score\",\"MutationTaster_pred\",\"MutationAssessor_score\",\"MutationAssessor_pred\",\"FATHMM_score\",\"FATHMM_pred\",\"RadialSVM_score\",\"RadialSVM_pred\",\"LR_score\",\"LR_pred\",\"VEST3_score\",\"CADD_raw\",\"CADD_phred\",\"GERP++_RS\",\"phyloP46way_placental\",\"phyloP100way_vertebrate\",\"SiPhy_29way_logOdds\",\"Interpro_domain\",\"SIFT_score.1\",\"SIFT_converted_rankscore\",\"SIFT_pred.1\",\"Polyphen2_HDIV_score.1\",\"Polyphen2_HDIV_rankscore\",\"Polyphen2_HDIV_pred.1\",\"Polyphen2_HVAR_score.1\",\"Polyphen2_HVAR_rankscore\",\"Polyphen2_HVAR_pred.1\",\"LRT_score.1\",\"LRT_converted_rankscore\",\"LRT_pred.1\",\"MutationTaster_score.1\",\"MutationTaster_converted_rankscore\",\"MutationTaster_pred.1\",\"MutationAssessor_score.1\",\"MutationAssessor_score_rankscore\",\"MutationAssessor_pred.1\",\"FATHMM_score.1\",\"FATHMM_converted_rankscore\",\"FATHMM_pred.1\",\"PROVEAN_score\",\"PROVEAN_converted_rankscore\",\"PROVEAN_pred\",\"VEST3_score.1\",\"VEST3_rankscore\",\"MetaSVM_score\",\"MetaSVM_rankscore\",\"MetaSVM_pred\",\"MetaLR_score\",\"MetaLR_rankscore\",\"MetaLR_pred\",\"M-CAP_score\",\"M-CAP_rankscore\",\"M-CAP_pred\",\"CADD_raw.1\",\"CADD_raw_rankscore\",\"CADD_phred.1\",\"DANN_score\",\"DANN_rankscore\",\"fathmm-MKL_coding_score\",\"fathmm-MKL_coding_rankscore\",\"fathmm-MKL_coding_pred\",\"Eigen_coding_or_noncoding\",\"Eigen-raw\",\"Eigen-PC-raw\",\"GenoCanyon_score\",\"GenoCanyon_score_rankscore\",\"integrated_fitCons_score\",\"integrated_fitCons_score_rankscore\",\"integrated_confidence_value\",\"GERP++_RS.1\",\"GERP++_RS_rankscore\",\"phyloP100way_vertebrate.1\",\"phyloP100way_vertebrate_rankscore\",\"phyloP20way_mammalian\",\"phyloP20way_mammalian_rankscore\",\"phastCons100way_vertebrate\",\"phastCons100way_vertebrate_rankscore\",\"phastCons20way_mammalian\",\"phastCons20way_mammalian_rankscore\",\"SiPhy_29way_logOdds.1\",\"SiPhy_29way_logOdds_rankscore\",\"Interpro_domain.1\",\"GTEx_V6_gene\",\"GTEx_V6_tissue\",\"dbscSNV_ADA_SCORE\",\"dbscSNV_RF_SCORE\",\"InterVar_automated\",\"PVS1\",\"PS1\",\"PS2\",\"PS3\",\"PS4\",\"PM1\",\"PM2\",\"PM3\",\"PM4\",\"PM5\",\"PM6\",\"PP1\",\"PP2\",\"PP3\",\"PP4\",\"PP5\",\"BA1\",\"BS1\",\"BS2\",\"BS3\",\"BS4\",\"BP1\",\"BP2\",\"BP3\",\"BP4\",\"BP5\",\"BP6\",\"BP7\",\"cosmic70\",\"MCAP\",\"REVEL\",\"regsnp_fpr\",\"regsnp_disease\",\"regsnp_splicing_site\"\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "pathogenic = pathogenic.loc[pathogenic['Ref'].isin(['A','T','G','C'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "pathogenic = pathogenic.loc[pathogenic['Alt'].isin(['A','T','G','C'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "tbfs = dd.read_csv(\"../data/input/tbfs/snp2tfbs_JASPAR_CORE_2014_vertwithout_chr.bed\",sep=\"\\t\",header=None,\n",
    "                   names=[\"Chr\",\"Start\",\"End\",\"Ref\",\"Alt\",\"rsid\",\"tfbs\",\"tfbsName\",\"tbfsScore\"], converters={i: str for i in range(250)})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>End</th>\n",
       "      <th>Start</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>rsid</th>\n",
       "      <th>tfbs</th>\n",
       "      <th>tfbsName</th>\n",
       "      <th>tbfsScore</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>10615</td>\n",
       "      <td>10616</td>\n",
       "      <td>CCGCCGTTGCAAAGGCGCGCCG</td>\n",
       "      <td>C</td>\n",
       "      <td>rs376342519</td>\n",
       "      <td>1</td>\n",
       "      <td>NRF1</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>11011</td>\n",
       "      <td>11012</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>rs544419019</td>\n",
       "      <td>2</td>\n",
       "      <td>PLAG1,EGR1</td>\n",
       "      <td>609,-186</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>13109</td>\n",
       "      <td>13110</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>rs540538026</td>\n",
       "      <td>3</td>\n",
       "      <td>ZNF263,PRDM1,SPIB</td>\n",
       "      <td>376,-9,0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>13115</td>\n",
       "      <td>13116</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>rs62635286</td>\n",
       "      <td>2</td>\n",
       "      <td>ZNF263,PRDM1</td>\n",
       "      <td>376,-9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>13117</td>\n",
       "      <td>13118</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>rs200579949</td>\n",
       "      <td>2</td>\n",
       "      <td>ZNF263,PRDM1</td>\n",
       "      <td>376,-9</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Chr    End  Start                     Ref Alt         rsid tfbs  \\\n",
       "0   1  10615  10616  CCGCCGTTGCAAAGGCGCGCCG   C  rs376342519    1   \n",
       "1   1  11011  11012                       C   G  rs544419019    2   \n",
       "2   1  13109  13110                       G   A  rs540538026    3   \n",
       "3   1  13115  13116                       T   G   rs62635286    2   \n",
       "4   1  13117  13118                       A   G  rs200579949    2   \n",
       "\n",
       "            tfbsName tbfsScore  \n",
       "0               NRF1        10  \n",
       "1         PLAG1,EGR1  609,-186  \n",
       "2  ZNF263,PRDM1,SPIB  376,-9,0  \n",
       "3       ZNF263,PRDM1    376,-9  \n",
       "4       ZNF263,PRDM1    376,-9  "
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tbfs.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1041582</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1041582</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1042136</td>\n",
       "      <td>T</td>\n",
       "      <td>TC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>1051775</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>1051775</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Chr      End Ref Alt\n",
       "0   1  1041582   C   T\n",
       "1   1  1041582   C   T\n",
       "2   1  1042136   T  TC\n",
       "3   1  1051775   G   A\n",
       "4   1  1051775   G   A"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pathogenic[[\"Chr\",\"End\",\"Ref\",\"Alt\"]].head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [],
   "source": [
    "pathogenic_tfbs = dd.merge(pathogenic,tbfs, how='left', on=['Chr', 'End', 'Ref','Alt']).compute()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>GC</th>\n",
       "      <th>CpG</th>\n",
       "      <th>motifECount</th>\n",
       "      <th>motifEHIPos</th>\n",
       "      <th>oAA</th>\n",
       "      <th>nAA</th>\n",
       "      <th>cDNApos</th>\n",
       "      <th>...</th>\n",
       "      <th>phyloP100way_vertebrate</th>\n",
       "      <th>SiPhy_29way_logOdds</th>\n",
       "      <th>PROVEAN_score</th>\n",
       "      <th>MetaSVM_score</th>\n",
       "      <th>MetaLR_score</th>\n",
       "      <th>integrated_fitCons_score</th>\n",
       "      <th>phyloP20way_mammalian</th>\n",
       "      <th>phastCons100way_vertebrate</th>\n",
       "      <th>phastCons20way_mammalian</th>\n",
       "      <th>tfbs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>0.384</td>\n",
       "      <td>0.027</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>0.57</td>\n",
       "      <td>0.08</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>A</td>\n",
       "      <td>P</td>\n",
       "      <td>738</td>\n",
       "      <td>...</td>\n",
       "      <td>1.470</td>\n",
       "      <td>8.506</td>\n",
       "      <td>-1.96</td>\n",
       "      <td>0.118</td>\n",
       "      <td>0.647</td>\n",
       "      <td>.</td>\n",
       "      <td>0.143</td>\n",
       "      <td>0.952</td>\n",
       "      <td>0.036</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>0.563</td>\n",
       "      <td>0.107</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>G</td>\n",
       "      <td>S</td>\n",
       "      <td>807</td>\n",
       "      <td>...</td>\n",
       "      <td>9.869</td>\n",
       "      <td>11.015</td>\n",
       "      <td>-5.81</td>\n",
       "      <td>1.133</td>\n",
       "      <td>0.950</td>\n",
       "      <td>.</td>\n",
       "      <td>1.048</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.942</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>0.576</td>\n",
       "      <td>0.067</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>K</td>\n",
       "      <td>E</td>\n",
       "      <td>625</td>\n",
       "      <td>...</td>\n",
       "      <td>1.127</td>\n",
       "      <td>12.576</td>\n",
       "      <td>-0.71</td>\n",
       "      <td>-0.925</td>\n",
       "      <td>0.101</td>\n",
       "      <td>.</td>\n",
       "      <td>1.199</td>\n",
       "      <td>0.002</td>\n",
       "      <td>1.000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>0.642</td>\n",
       "      <td>0.093</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>G</td>\n",
       "      <td>V</td>\n",
       "      <td>962</td>\n",
       "      <td>...</td>\n",
       "      <td>7.546</td>\n",
       "      <td>16.019</td>\n",
       "      <td>-5.73</td>\n",
       "      <td>-0.668</td>\n",
       "      <td>0.195</td>\n",
       "      <td>.</td>\n",
       "      <td>0.953</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.914</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 94 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  Ref Alt     Consequence     GC    CpG motifECount motifEHIPos oAA nAA  \\\n",
       "0   A   T        INTRONIC  0.384  0.027          NA          NA  NA  NA   \n",
       "1   G   C  NON_SYNONYMOUS   0.57   0.08          NA          NA   A   P   \n",
       "2   G   A  NON_SYNONYMOUS  0.563  0.107          NA          NA   G   S   \n",
       "3   A   G  NON_SYNONYMOUS  0.576  0.067          NA          NA   K   E   \n",
       "4   G   T  NON_SYNONYMOUS  0.642  0.093          NA          NA   G   V   \n",
       "\n",
       "  cDNApos  ... phyloP100way_vertebrate SiPhy_29way_logOdds PROVEAN_score  \\\n",
       "0      NA  ...                       .                   .             .   \n",
       "1     738  ...                   1.470               8.506         -1.96   \n",
       "2     807  ...                   9.869              11.015         -5.81   \n",
       "3     625  ...                   1.127              12.576         -0.71   \n",
       "4     962  ...                   7.546              16.019         -5.73   \n",
       "\n",
       "  MetaSVM_score MetaLR_score integrated_fitCons_score phyloP20way_mammalian  \\\n",
       "0             .            .                        .                     .   \n",
       "1         0.118        0.647                        .                 0.143   \n",
       "2         1.133        0.950                        .                 1.048   \n",
       "3        -0.925        0.101                        .                 1.199   \n",
       "4        -0.668        0.195                        .                 0.953   \n",
       "\n",
       "  phastCons100way_vertebrate phastCons20way_mammalian tfbs  \n",
       "0                          .                        .  NaN  \n",
       "1                      0.952                    0.036  NaN  \n",
       "2                      1.000                    0.942  NaN  \n",
       "3                      0.002                    1.000  NaN  \n",
       "4                      1.000                    0.914  NaN  \n",
       "\n",
       "[5 rows x 94 columns]"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Chr     End     Ref     Alt     Type    Length  AnnoType        Consequence     \n",
    "#ConsScore       ConsDetail      GC      CpmotifECount      motifEName      \n",
    "#motifEHIPos     motifEScoreChng oAA     nAA     GeneID  FeatureID       GeneName        \n",
    "#CCDS       Intron  Exon    cDNApos relcDNApos      CDSpos  relCDSpos       protPos relProtPos     \n",
    "#Domain  Dst2Splice      Dst2SplType        minDistTSS      minDistTSE      SIFTcat SIFTval \n",
    "#PolyPhenCat     PolyPhenVal     priPhCons       mamPhCons verPhCons        priPhyloP       \n",
    "#mamPhyloP       verPhyloP       bStatistic      targetScan      mirSVR-Score    mirSVR-E  \n",
    "#mirSVR-Aln       cHmm_E1 cHmm_E2 cHmm_E3 cHmm_E4 cHmm_E5 cHmm_E6 cHmm_E7 cHmm_E8 cHmm_E9 \n",
    "#cHmm_E10        cHmm_E11        cHmm_E12   cHmm_E13        cHmm_E14        cHmm_E15        \n",
    "#cHmm_E16        cHmm_E17        cHmm_E18        cHmm_E19        cHmm_E20   cHmm_E21        \n",
    "#cHmm_E22        cHmm_E23        cHmm_E24        cHmm_E25        GerpRS  GerpRSpval      GerpN   \n",
    "#GerpS      tOverlapMotifs  motifDist       EncodeH3K4me1-sum       EncodeH3K4me1-max       \n",
    "#EncodeH3K4me2-sum       EncodeH3K4me2-max  EncodeH3K4me3-sum       EncodeH3K4me3-max       \n",
    "#EncodeH3K9ac-sum        EncodeH3K9ac-max        EncodeH3K9me3-sum EncodeH3K9me3-max        \n",
    "#EncodeH3K27ac-sum       EncodeH3K27ac-max       EncodeH3K27me3-sum      EncodeH3K27me3-max      \n",
    "#EncodeH3K36me3-sum EncodeH3K36me3-max      EncodeH3K79me2-sum      EncodeH3K79me2-max      \n",
    "#EncodeH4K20me1-sum      EncodeH4K20me1-max EncodeH2AFZ-sum EncodeH2AFZ-max EncodeDNase-sum \n",
    "#EncodeDNase-max EncodetotalRNA-sum      EncodetotalRNA-max      Grantham   Dist2Mutation   Freq100bp       \n",
    "#Rare100bp       Sngl100bp       Freq1000bp      Rare1000bp      Sngl1000bp      Freq10000bp        \n",
    "#Rare10000bp     Sngl10000bp     EnsembleRegulatoryFeature       dbscSNV-ada_score       dbscSNV-rf_score  \n",
    "#RemapOverlapTF   RemapOverlapCRawScore   PHRED   Start   Func.refGene    Gene.refGene    GeneDetail.refGene      \n",
    "#ExonicFunc.refGene AAChange.refGene        Func.knownGene  Gene.knownGene  GeneDetail.knownGene    \n",
    "#ExonicFunc.knownGene    AAChange.knownGene Func.ensGene    Gene.ensGene    GeneDetail.ensGene      \n",
    "#ExonicFunc.ensGene      AAChange.ensGene        SIFT_scoreSIFT_pred        Polyphen2_HDIV_score    \n",
    "#Polyphen2_HDIV_pred     Polyphen2_HVAR_score    Polyphen2_HVAR_pred     LRT_score LRT_pred MutationTaster_score    \n",
    "#MutationTaster_pred     MutationAssessor_score  MutationAssessor_pred   FATHMM_score    FATHMM_pred        \n",
    "#RadialSVM_score RadialSVM_pred  LR_score        LR_pred VEST3_score     CADD_raw        CADD_phred      \n",
    "#GERP++_RS  phyloP46way_placental   phyloP100way_vertebrate SiPhy_29way_logOdds     Interpro_domain SIFT_score.1    \n",
    "#SIFT_converted_rankscore   SIFT_pred.1     Polyphen2_HDIV_score.1  Polyphen2_HDIV_rankscore        \n",
    "#Polyphen2_HDIV_pred.1   Polyphen2_HVAR_score.1     Polyphen2_HVAR_rankscore        Polyphen2_HVAR_pred.1   \n",
    "#LRT_score.1     LRT_converted_rankscore LRT_pred.1MutationTaster_score.1   MutationTaster_converted_rankscore      \n",
    "#MutationTaster_pred.1   MutationAssessor_score.1        MutationAssessor_score_rankscore   \n",
    "#MutationAssessor_pred.1 FATHMM_score.1  FATHMM_converted_rankscore      FATHMM_pred.1   PROVEAN_score      \n",
    "#PROVEAN_converted_rankscore     PROVEAN_pred    VEST3_score.1   VEST3_rankscore MetaSVM_score   MetaSVM_rankscore  \n",
    "#MetaSVM_pred    MetaLR_score    MetaLR_rankscore        MetaLR_pred     M-CAP_score     M-CAP_rankscore \n",
    "#M-CAP_predCADD_raw.1       CADD_raw_rankscore      CADD_phred.1    DANN_score      DANN_rankscore  \n",
    "#fathmm-MKL_coding_score fathmm-MKL_coding_rankscore        fathmm-MKL_coding_pred  Eigen_coding_or_noncoding       \n",
    "#Eigen-raw       Eigen-PC-raw    GenoCanyon_score   GenoCanyon_score_rankscore      integrated_fitCons_score        \n",
    "#integrated_fitCons_score_rankscore      integrated_confidence_value        GERP++_RS.1     GERP++_RS_rankscore     \n",
    "#phyloP100way_vertebrate.1       phyloP100way_vertebrate_rankscore  phyloP20way_mammalian   \n",
    "#phyloP20way_mammalian_rankscore phastCons100way_vertebrate      phastCons100way_vertebrate_rankscore       \n",
    "#phastCons20way_mammalian        phastCons20way_mammalian_rankscore      SiPhy_29way_logOdds.1   \n",
    "#SiPhy_29way_logOdds_rankscore      Interpro_domain.1       GTEx_V6_gene    GTEx_V6_tissue  dbscSNV_ADA_SCORE       \n",
    "#dbscSNV_RF_SCORE  InterVar_automated       PVS1    PS1     PS2     PS3     PS4     PM1     PM2     PM3     \n",
    "#PM4     PM5     PM6     PP1     PPPP3      PP4     PP5     BA1     BS1     BS2     BS3     BS4     BP1     \n",
    "#BP2     BP3     BP4     BP5     BP6     BP7     cosmic70   MCAP    REVEL   regsnp_fpr      regsnp_disease  \n",
    "#regsnp_splicing_site\n",
    "\n",
    "case = pathogenic_tfbs[[\"Ref\",\"Alt\",\"Consequence\",\"GC\",\"CpG\",\"motifECount\",\"motifEHIPos\",\"oAA\",\"nAA\",\"cDNApos\",\n",
    "           \"relcDNApos\",\"CDSpos\",\"relCDSpos\",\"protPos\",\"relProtPos\",\"Domain\",\"Dst2Splice\",\"Dst2SplType\",\n",
    "           \"minDistTSS\",\"minDistTSE\",\"SIFTcat\",\"SIFTval\",\"PolyPhenCat\",\"PolyPhenVal\",\"priPhCons\",\"mamPhCons\",\n",
    "           \"verPhCons\",\"priPhyloP\",\"mamPhyloP\",\"verPhyloP\",\"bStatistic\",\"cHmm_E1\",\"cHmm_E2\",\"cHmm_E3\",\"cHmm_E4\",\n",
    "           \"cHmm_E5\",\"cHmm_E6\",\"cHmm_E7\",\"cHmm_E8\",\"cHmm_E9\",\"cHmm_E10\",\"cHmm_E11\",\"cHmm_E12\",\"cHmm_E13\",\n",
    "           \"cHmm_E14\",\"cHmm_E15\",\"cHmm_E16\",\"cHmm_E17\",\"cHmm_E18\",\"cHmm_E19\",\"cHmm_E20\",\"cHmm_E21\",\"cHmm_E22\",\n",
    "           \"cHmm_E23\",\"cHmm_E24\",\"cHmm_E25\",\"GerpRS\",\"GerpRSpval\",\"GerpN\",\"GerpS\",\"tOverlapMotifs\",\"motifDist\",\n",
    "           \"EncodeH3K4me1-max\",\"EncodeH3K4me2-max\",\"EncodeH3K4me3-max\",\"EncodeH3K9ac-max\",\"EncodeH3K9me3-max\",\n",
    "           \"EncodeH3K27ac-max\",\"EncodeH3K27me3-max\",\"EncodeH3K36me3-max\",\"EncodeH3K79me2-max\",\"EncodeH4K20me1-max\",\n",
    "           \"EncodeH2AFZ-max\",\"EncodeDNase-max\",\"EncodetotalRNA-max\",\"Grantham\",\"SIFT_score\",\"Polyphen2_HDIV_score\",\n",
    "           \"Polyphen2_HVAR_score\",\"LRT_score\",\"MutationTaster_score\",\"MutationAssessor_score\",\"LR_score\",\"GERP++_RS\",\n",
    "           \"phyloP100way_vertebrate\",\"SiPhy_29way_logOdds\",\"PROVEAN_score\",\"MetaSVM_score\",\"MetaLR_score\",\n",
    "           \"integrated_fitCons_score\",\"phyloP20way_mammalian\",\"phastCons100way_vertebrate\",\n",
    "           \"phastCons20way_mammalian\",\"tfbs\"]]\n",
    "\n",
    "\n",
    "case.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>GC</th>\n",
       "      <th>CpG</th>\n",
       "      <th>motifECount</th>\n",
       "      <th>motifEHIPos</th>\n",
       "      <th>oAA</th>\n",
       "      <th>nAA</th>\n",
       "      <th>cDNApos</th>\n",
       "      <th>...</th>\n",
       "      <th>phyloP100way_vertebrate</th>\n",
       "      <th>SiPhy_29way_logOdds</th>\n",
       "      <th>PROVEAN_score</th>\n",
       "      <th>MetaSVM_score</th>\n",
       "      <th>MetaLR_score</th>\n",
       "      <th>integrated_fitCons_score</th>\n",
       "      <th>phyloP20way_mammalian</th>\n",
       "      <th>phastCons100way_vertebrate</th>\n",
       "      <th>phastCons20way_mammalian</th>\n",
       "      <th>tfbs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>0.384</td>\n",
       "      <td>0.027</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>0.57</td>\n",
       "      <td>0.08</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>P</td>\n",
       "      <td>738</td>\n",
       "      <td>...</td>\n",
       "      <td>1.470</td>\n",
       "      <td>8.506</td>\n",
       "      <td>-1.96</td>\n",
       "      <td>0.118</td>\n",
       "      <td>0.647</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.143</td>\n",
       "      <td>0.952</td>\n",
       "      <td>0.036</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>0.563</td>\n",
       "      <td>0.107</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>S</td>\n",
       "      <td>807</td>\n",
       "      <td>...</td>\n",
       "      <td>9.869</td>\n",
       "      <td>11.015</td>\n",
       "      <td>-5.81</td>\n",
       "      <td>1.133</td>\n",
       "      <td>0.950</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.048</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.942</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>0.576</td>\n",
       "      <td>0.067</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>K</td>\n",
       "      <td>E</td>\n",
       "      <td>625</td>\n",
       "      <td>...</td>\n",
       "      <td>1.127</td>\n",
       "      <td>12.576</td>\n",
       "      <td>-0.71</td>\n",
       "      <td>-0.925</td>\n",
       "      <td>0.101</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.199</td>\n",
       "      <td>0.002</td>\n",
       "      <td>1.000</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>NON_SYNONYMOUS</td>\n",
       "      <td>0.642</td>\n",
       "      <td>0.093</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>G</td>\n",
       "      <td>V</td>\n",
       "      <td>962</td>\n",
       "      <td>...</td>\n",
       "      <td>7.546</td>\n",
       "      <td>16.019</td>\n",
       "      <td>-5.73</td>\n",
       "      <td>-0.668</td>\n",
       "      <td>0.195</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.953</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.914</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 94 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  Ref Alt     Consequence     GC    CpG motifECount motifEHIPos  oAA  nAA  \\\n",
       "0   A   T        INTRONIC  0.384  0.027         NaN         NaN  NaN  NaN   \n",
       "1   G   C  NON_SYNONYMOUS   0.57   0.08         NaN         NaN    A    P   \n",
       "2   G   A  NON_SYNONYMOUS  0.563  0.107         NaN         NaN    G    S   \n",
       "3   A   G  NON_SYNONYMOUS  0.576  0.067         NaN         NaN    K    E   \n",
       "4   G   T  NON_SYNONYMOUS  0.642  0.093         NaN         NaN    G    V   \n",
       "\n",
       "  cDNApos  ... phyloP100way_vertebrate SiPhy_29way_logOdds PROVEAN_score  \\\n",
       "0     NaN  ...                     NaN                 NaN           NaN   \n",
       "1     738  ...                   1.470               8.506         -1.96   \n",
       "2     807  ...                   9.869              11.015         -5.81   \n",
       "3     625  ...                   1.127              12.576         -0.71   \n",
       "4     962  ...                   7.546              16.019         -5.73   \n",
       "\n",
       "  MetaSVM_score MetaLR_score integrated_fitCons_score phyloP20way_mammalian  \\\n",
       "0           NaN          NaN                      NaN                   NaN   \n",
       "1         0.118        0.647                      NaN                 0.143   \n",
       "2         1.133        0.950                      NaN                 1.048   \n",
       "3        -0.925        0.101                      NaN                 1.199   \n",
       "4        -0.668        0.195                      NaN                 0.953   \n",
       "\n",
       "  phastCons100way_vertebrate phastCons20way_mammalian tfbs  \n",
       "0                        NaN                      NaN  NaN  \n",
       "1                      0.952                    0.036  NaN  \n",
       "2                      1.000                    0.942  NaN  \n",
       "3                      0.002                    1.000  NaN  \n",
       "4                      1.000                    0.914  NaN  \n",
       "\n",
       "[5 rows x 94 columns]"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#replace ., * , null with NaN \n",
    "#case = case.replace('^.$', np.nan)\n",
    "#case = case.replace('*', np.nan)\n",
    "\n",
    "\n",
    "case = case.replace('null', np.nan, regex=True)\n",
    "case = case.replace('Null', np.nan, regex=True)\n",
    "case = case.replace('NA', np.nan, regex=True)\n",
    "case = case.replace('.', np.nan)\n",
    "case = case.replace('-', np.nan)\n",
    "case = case.replace('*', np.nan)\n",
    "\n",
    "case.to_csv(\"../analysis/trainingDataSet/slected_pathogenic.csv\", sep=\"\\t\",index=False)\n",
    "case.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(113485, 94)"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "case.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Type</th>\n",
       "      <th>Length</th>\n",
       "      <th>AnnoType</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>ConsScore</th>\n",
       "      <th>ConsDetail</th>\n",
       "      <th>...</th>\n",
       "      <th>BP4</th>\n",
       "      <th>BP5</th>\n",
       "      <th>BP6</th>\n",
       "      <th>BP7</th>\n",
       "      <th>cosmic70</th>\n",
       "      <th>MCAP</th>\n",
       "      <th>REVEL</th>\n",
       "      <th>regsnp_fpr</th>\n",
       "      <th>regsnp_disease</th>\n",
       "      <th>regsnp_splicing_site</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>78208116</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>16</td>\n",
       "      <td>78215334</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>16</td>\n",
       "      <td>78220722</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>16</td>\n",
       "      <td>78221967</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>REGULATORY</td>\n",
       "      <td>4</td>\n",
       "      <td>regulatory</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>16</td>\n",
       "      <td>78221967</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>SNV</td>\n",
       "      <td>0</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>2</td>\n",
       "      <td>intron</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 270 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  Chr       End Ref Alt Type Length           AnnoType Consequence ConsScore  \\\n",
       "0  16  78208116   G   A  SNV      0         Transcript    INTRONIC         2   \n",
       "1  16  78215334   T   C  SNV      0         Transcript    INTRONIC         2   \n",
       "2  16  78220722   A   G  SNV      0         Transcript    INTRONIC         2   \n",
       "3  16  78221967   G   C  SNV      0  RegulatoryFeature  REGULATORY         4   \n",
       "4  16  78221967   G   C  SNV      0         Transcript    INTRONIC         2   \n",
       "\n",
       "   ConsDetail  ... BP4 BP5 BP6 BP7 cosmic70 MCAP REVEL regsnp_fpr  \\\n",
       "0      intron  ...   .   .   .   .        .    .     .          .   \n",
       "1      intron  ...   .   .   .   .        .    .     .          .   \n",
       "2      intron  ...   .   .   .   .        .    .     .          .   \n",
       "3  regulatory  ...   .   .   .   .        .    .     .          .   \n",
       "4      intron  ...   .   .   .   .        .    .     .          .   \n",
       "\n",
       "  regsnp_disease regsnp_splicing_site  \n",
       "0              .                    .  \n",
       "1              .                    .  \n",
       "2              .                    .  \n",
       "3              .                    .  \n",
       "4              .                    .  \n",
       "\n",
       "[5 rows x 270 columns]"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "##human Drive \n",
    "\n",
    "human_drive = pd.read_csv(\"../analysis/trainingDataSet/human_Drive_cadd_annovar_annotation_uniqe.tsv\",sep=\"\\t\", converters={i: str for i in range(250)})\n",
    "human_drive.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "human_drive = human_drive.loc[human_drive['Ref'].isin(['A','T','G','C'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "human_drive = human_drive.loc[human_drive['Alt'].isin(['A','T','G','C'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "human_drive_tfbs = dd.merge(human_drive,tbfs, how='left', on=['Chr', 'End', 'Ref','Alt']).compute()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>GC</th>\n",
       "      <th>CpG</th>\n",
       "      <th>motifECount</th>\n",
       "      <th>motifEHIPos</th>\n",
       "      <th>oAA</th>\n",
       "      <th>nAA</th>\n",
       "      <th>cDNApos</th>\n",
       "      <th>...</th>\n",
       "      <th>phyloP100way_vertebrate</th>\n",
       "      <th>SiPhy_29way_logOdds</th>\n",
       "      <th>PROVEAN_score</th>\n",
       "      <th>MetaSVM_score</th>\n",
       "      <th>MetaLR_score</th>\n",
       "      <th>integrated_fitCons_score</th>\n",
       "      <th>phyloP20way_mammalian</th>\n",
       "      <th>phastCons100way_vertebrate</th>\n",
       "      <th>phastCons20way_mammalian</th>\n",
       "      <th>tfbs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>0.503</td>\n",
       "      <td>0.04</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>0.53</td>\n",
       "      <td>0.013</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>0.371</td>\n",
       "      <td>0.04</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>INTRONIC</td>\n",
       "      <td>0.391</td>\n",
       "      <td>0.013</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>DOWNSTREAM</td>\n",
       "      <td>0.311</td>\n",
       "      <td>0.013</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>NA</td>\n",
       "      <td>...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 94 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  Ref Alt Consequence     GC    CpG motifECount motifEHIPos oAA nAA cDNApos  \\\n",
       "0   G   A    INTRONIC  0.503   0.04          NA          NA  NA  NA      NA   \n",
       "1   A   C    INTRONIC   0.53  0.013          NA          NA  NA  NA      NA   \n",
       "2   T   C    INTRONIC  0.371   0.04          NA          NA  NA  NA      NA   \n",
       "3   A   G    INTRONIC  0.391  0.013          NA          NA  NA  NA      NA   \n",
       "4   T   C  DOWNSTREAM  0.311  0.013          NA          NA  NA  NA      NA   \n",
       "\n",
       "   ... phyloP100way_vertebrate SiPhy_29way_logOdds PROVEAN_score  \\\n",
       "0  ...                       .                   .             .   \n",
       "1  ...                       .                   .             .   \n",
       "2  ...                       .                   .             .   \n",
       "3  ...                       .                   .             .   \n",
       "4  ...                       .                   .             .   \n",
       "\n",
       "  MetaSVM_score MetaLR_score integrated_fitCons_score phyloP20way_mammalian  \\\n",
       "0             .            .                        .                     .   \n",
       "1             .            .                        .                     .   \n",
       "2             .            .                        .                     .   \n",
       "3             .            .                        .                     .   \n",
       "4             .            .                        .                     .   \n",
       "\n",
       "  phastCons100way_vertebrate phastCons20way_mammalian tfbs  \n",
       "0                          .                        .  NaN  \n",
       "1                          .                        .  NaN  \n",
       "2                          .                        .  NaN  \n",
       "3                          .                        .  NaN  \n",
       "4                          .                        .  NaN  \n",
       "\n",
       "[5 rows x 94 columns]"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "control = human_drive_tfbs[[\"Ref\",\"Alt\",\"Consequence\",\"GC\",\"CpG\",\"motifECount\",\"motifEHIPos\",\"oAA\",\"nAA\",\"cDNApos\",\n",
    "           \"relcDNApos\",\"CDSpos\",\"relCDSpos\",\"protPos\",\"relProtPos\",\"Domain\",\"Dst2Splice\",\"Dst2SplType\",\n",
    "           \"minDistTSS\",\"minDistTSE\",\"SIFTcat\",\"SIFTval\",\"PolyPhenCat\",\"PolyPhenVal\",\"priPhCons\",\"mamPhCons\",\n",
    "           \"verPhCons\",\"priPhyloP\",\"mamPhyloP\",\"verPhyloP\",\"bStatistic\",\"cHmm_E1\",\"cHmm_E2\",\"cHmm_E3\",\"cHmm_E4\",\n",
    "           \"cHmm_E5\",\"cHmm_E6\",\"cHmm_E7\",\"cHmm_E8\",\"cHmm_E9\",\"cHmm_E10\",\"cHmm_E11\",\"cHmm_E12\",\"cHmm_E13\",\n",
    "           \"cHmm_E14\",\"cHmm_E15\",\"cHmm_E16\",\"cHmm_E17\",\"cHmm_E18\",\"cHmm_E19\",\"cHmm_E20\",\"cHmm_E21\",\"cHmm_E22\",\n",
    "           \"cHmm_E23\",\"cHmm_E24\",\"cHmm_E25\",\"GerpRS\",\"GerpRSpval\",\"GerpN\",\"GerpS\",\"tOverlapMotifs\",\"motifDist\",\n",
    "           \"EncodeH3K4me1-max\",\"EncodeH3K4me2-max\",\"EncodeH3K4me3-max\",\"EncodeH3K9ac-max\",\"EncodeH3K9me3-max\",\n",
    "           \"EncodeH3K27ac-max\",\"EncodeH3K27me3-max\",\"EncodeH3K36me3-max\",\"EncodeH3K79me2-max\",\"EncodeH4K20me1-max\",\n",
    "           \"EncodeH2AFZ-max\",\"EncodeDNase-max\",\"EncodetotalRNA-max\",\"Grantham\",\"SIFT_score\",\"Polyphen2_HDIV_score\",\n",
    "           \"Polyphen2_HVAR_score\",\"LRT_score\",\"MutationTaster_score\",\"MutationAssessor_score\",\"LR_score\",\"GERP++_RS\",\n",
    "           \"phyloP100way_vertebrate\",\"SiPhy_29way_logOdds\",\"PROVEAN_score\",\"MetaSVM_score\",\"MetaLR_score\",\n",
    "           \"integrated_fitCons_score\",\"phyloP20way_mammalian\",\"phastCons100way_vertebrate\",\n",
    "           \"phastCons20way_mammalian\",\"tfbs\"]]\n",
    "\n",
    "\n",
    "control.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [],
   "source": [
    "#replace ., * , null with NaN \n",
    "#control = control.replace('.', np.nan, regex=True)\n",
    "#control = control.replace('*', np.nan, regex=True)\n",
    "control = control.replace('null', np.nan, regex=True)\n",
    "control = control.replace('Null', np.nan, regex=True)\n",
    "control = control.replace('NA', np.nan, regex=True)\n",
    "control = control.replace('.', np.nan)\n",
    "control = control.replace('-', np.nan)\n",
    "control = control.replace('*', np.nan)\n",
    "control.to_csv(\"../analysis/trainingDataSet/slected_human_Drive.csv\", sep=\"\\t\",index=False)\n",
    "#control.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1674865, 94)"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "control.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  },
  "notify_time": "0"
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
